--- fontconfig-2.13.0/m4/ac_check_symbol.m4
+++ fontconfig-2.13.0/m4/ac_check_symbol.m4
@@ -1,6 +1,6 @@
 dnl @synopsis AC_CHECK_SYMBOL(SYMBOL, HEADER... [,ACTION-IF-FOUND [,ACTION-IF-NOT-FOUND]])
 dnl
-dnl a wrapper around AC_EGREP_HEADER the shellvar $ac_found will hold
+dnl a wrapper around AC_EGREP_HEADER the shellvar $ac_cv_func_$1 will hold
 dnl the HEADER-name that had been containing the symbol. This value is
 dnl shown to the user.
 dnl
@@ -16,24 +16,24 @@ AC_CACHE_VAL(ac_cv_func_$1,
 changequote(, )dnl
 symbol="[^a-zA-Z_0-9]$1[^a-zA-Z_0-9]"
 changequote([, ])dnl
-ac_found=no
+ac_cv_func_$1=no
 for ac_header in $2 ; do
   ac_safe=`echo "$ac_header" | sed 'y%./+-%__p_%' `
-  if test $ac_found != "yes" ; then
-      if eval "test \"`echo '$ac_cv_header_'$ac_safe`\" = yes"; then
-            AC_EGREP_HEADER( $symbol, $ac_header, [ac_found="$ac_header"] )
-      fi
+  if test $ac_cv_func_$1 = "no" ; then
+    if eval "test \"`echo '$ac_cv_header_'$ac_safe`\" = yes"; then
+      AC_EGREP_HEADER( $symbol, $ac_header, [ac_cv_func_$1="$ac_header"] )
+    fi
   fi
-done
-if test "$ac_found" != "no" ; then
-  AC_MSG_RESULT($ac_found)
+done])
+if test "$ac_cv_func_$1" != "no" ; then
+  AC_MSG_RESULT($ac_cv_func_$1)
   ifelse([$3], , :, [$3])
 else
   AC_MSG_RESULT(no)
   ifelse([$4], , , [$4
 ])dnl
 fi
-])])
+])
 
 dnl AC_CHECK_SYMBOLS( symbol..., header... [, action-if-found [, action-if-not-found]])
 AC_DEFUN([AC_CHECK_SYMBOLS],
