diff --git a/bglibs-1.106/net/bindu.c b/bglibs-fixed/net/bindu.c
index bf228dd..f039182 100644
--- a/bglibs-1.106/net/bindu.c
+++ b/bglibs-fixed/net/bindu.c
@@ -16,6 +16,7 @@
  * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301 USA
  */
 #include <errno.h>
+#include <string.h>
 #include <sys/types.h>
 #include <sys/param.h>
 #include <sys/socket.h>
diff --git a/bglibs-1.106/net/connectu.c b/bglibs-fixed/net/connectu.c
index a0a00ad..9f2944a 100644
--- a/bglibs-1.106/net/connectu.c
+++ b/bglibs-fixed/net/connectu.c
@@ -16,6 +16,7 @@
  * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301 USA
  */
 #include <errno.h>
+#include <string.h>
 #include <sys/types.h>
 #include <sys/param.h>
 #include <sys/socket.h>
