From 9b7fec2eb0c2a8eaf574a80b18fe7bb6612a6b37 Mon Sep 17 00:00:00 2001
From: Samuel Holland <samuel@sholland.org>
Date: Sat, 13 Oct 2018 01:14:58 +0000
Subject: [PATCH 6/9] powerpc: Add missing check in struct alignment

---
 src/powerpc/ffi_linux64.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/powerpc/ffi_linux64.c b/src/powerpc/ffi_linux64.c
index 197a270..d755c71 100644
--- a/src/powerpc/ffi_linux64.c
+++ b/src/powerpc/ffi_linux64.c
@@ -552,7 +552,11 @@ ffi_prep_args64 (extended_cif *ecif, unsigned long *const stack)
 	      if (align > 16)
 		align = 16;
 	      if (align > 1)
-		next_arg.p = FFI_ALIGN (next_arg.p, align);
+                {
+                  next_arg.p = FFI_ALIGN (next_arg.p, align);
+                  if (next_arg.ul == gpr_end.ul)
+                    next_arg.ul = rest.ul;
+                }
 	    }
 	  elt = discover_homogeneous_aggregate (*ptr, &elnum);
 	  if (elt)
-- 
2.21.0

