--- iproute2-4.17.0/bridge/monitor.c
+++ iproute2-4.17.0/bridge/monitor.c
@@ -27,7 +27,7 @@
 
 
 static void usage(void) __attribute__((noreturn));
-int prefix_banner;
+static int prefix_banner;
 
 static void usage(void)
 {
--- iproute2-4.17.0/ip/ipmonitor.c
+++ iproute2-4.17.0/ip/ipmonitor.c
@@ -24,8 +24,8 @@
 #include "ip_common.h"
 
 static void usage(void) __attribute__((noreturn));
-int prefix_banner;
-int listen_all_nsid;
+static int prefix_banner;
+static int listen_all_nsid;
 
 static void usage(void)
 {
--- iproute2-4.17.0/ip/ipmroute.c
+++ iproute2-4.17.0/ip/ipmroute.c
@@ -44,7 +44,7 @@ static void usage(void)
 	exit(-1);
 }
 
-struct rtfilter {
+static struct rtfilter {
 	int tb;
 	int af;
 	int iif;
--- iproute2-4.17.0/ip/xfrm_monitor.c
+++ iproute2-4.17.0/ip/xfrm_monitor.c
@@ -34,7 +34,7 @@
 #include "ip_common.h"
 
 static void usage(void) __attribute__((noreturn));
-int listen_all_nsid;
+static int listen_all_nsid;
 
 static void usage(void)
 {
