--- acpica-unix-20160729/source/tools/acpiexec/aemain.c
+++ acpica-unix-20160729/source/tools/acpiexec/aemain.c
@@ -157,7 +157,6 @@ AcpiDbRunBatchMode (
 
 UINT8                       AcpiGbl_RegionFillValue = 0;
 BOOLEAN                     AcpiGbl_IgnoreErrors = FALSE;
-BOOLEAN                     AcpiGbl_DbOpt_NoRegionSupport = FALSE;
 UINT8                       AcpiGbl_UseHwReducedFadt = FALSE;
 BOOLEAN                     AcpiGbl_DoInterfaceTests = FALSE;
 BOOLEAN                     AcpiGbl_LoadTestTables = FALSE;
--- acpica-unix-20160729/source/components/debugger/dbxface.c
+++ acpica-unix-20160729/source/components/debugger/dbxface.c
@@ -538,6 +538,7 @@ AcpiInitializeDebugger (
     AcpiGbl_DbOutputFlags       = ACPI_DB_CONSOLE_OUTPUT;
 
     AcpiGbl_DbOpt_NoIniMethods  = FALSE;
+    AcpiGbl_DbOpt_NoRegionSupport = FALSE;
 
     AcpiGbl_DbBuffer = AcpiOsAllocate (ACPI_DEBUG_BUFFER_SIZE);
     if (!AcpiGbl_DbBuffer)
--- acpica-unix-20160729/source/compiler/prparser.l
+++ acpica-unix-20160729/source/compiler/prparser.l
@@ -188,7 +188,7 @@ Identifier      [a-zA-Z][0-9a-zA-Z]*
 /*
  * Local support functions
  */
-YY_BUFFER_STATE         LexBuffer;
+static YY_BUFFER_STATE         LexBuffer;
 
 
 /******************************************************************************
--- acpica-unix-20160729/source/compiler/aslcompiler.l
+++ acpica-unix-20160729/source/compiler/aslcompiler.l
@@ -119,7 +119,6 @@
 
 #include <stdlib.h>
 #include <string.h>
-YYSTYPE AslCompilerlval;
 
 /*
  * Generation:  Use the following command line:
--- acpica-unix-20160729/source/compiler/dtparser.l
+++ acpica-unix-20160729/source/compiler/dtparser.l
@@ -172,7 +172,7 @@ NewLine         [\n]
 /*
  * Local support functions
  */
-YY_BUFFER_STATE         LexBuffer;
+static YY_BUFFER_STATE         LexBuffer;
 
 /******************************************************************************
  *
