diff -ur criu-2.0/criu/fsnotify.c criu-2.0-fixed/criu/fsnotify.c
--- criu-2.0/criu/fsnotify.c	2016-03-07 11:56:59.000000000 -0600
+++ criu-2.0-fixed/criu/fsnotify.c	2016-03-26 12:14:15.043544024 -0500
@@ -103,7 +103,7 @@
 
 static int open_by_handle(void *arg, int fd, int pid)
 {
-	return open_by_handle_at(fd, arg, O_PATH);
+	return syscall(SYS_open_by_handle_at, fd, arg, O_PATH);
 }
 
 static char *alloc_openable(unsigned int s_dev, unsigned long i_ino, FhEntry *f_handle)
diff -ur criu-2.0/criu/include/files.h criu-2.0-fixed/criu/include/files.h
--- criu-2.0/criu/include/files.h	2016-03-07 11:56:59.000000000 -0600
+++ criu-2.0-fixed/criu/include/files.h	2016-03-26 11:34:25.570911756 -0500
@@ -1,6 +1,8 @@
 #ifndef __CR_FILES_H__
 #define __CR_FILES_H__
 
+#include <sys/stat.h>
+
 #include "compiler.h"
 #include "asm/types.h"
 #include "fcntl.h"
diff -ur criu-2.0/criu/include/syscall-types.h criu-2.0-fixed/criu/include/syscall-types.h
--- criu-2.0/criu/include/syscall-types.h	2016-03-07 11:56:59.000000000 -0600
+++ criu-2.0-fixed/criu/include/syscall-types.h	2016-03-26 11:36:59.190081730 -0500
@@ -8,6 +8,7 @@
 #ifndef __CR_SYSCALL_TYPES_H__
 #define __CR_SYSCALL_TYPES_H__
 
+#include <fcntl.h>
 #include <sys/time.h>
 #include <arpa/inet.h>
 #include <sched.h>
diff -ur criu-2.0/criu/pie/restorer.c criu-2.0-fixed/criu/pie/restorer.c
--- criu-2.0/criu/pie/restorer.c	2016-03-07 11:56:59.000000000 -0600
+++ criu-2.0-fixed/criu/pie/restorer.c	2016-03-26 11:38:25.054902109 -0500
@@ -3,6 +3,7 @@
 
 #include <linux/securebits.h>
 #include <linux/capability.h>
+#include <sys/file.h>
 #include <sys/types.h>
 #include <sys/mman.h>
 #include <sys/stat.h>
diff -ur criu-2.0/lib/c/criu.c criu-2.0-fixed/lib/c/criu.c
--- criu-2.0/lib/c/criu.c	2016-03-07 11:56:59.000000000 -0600
+++ criu-2.0-fixed/lib/c/criu.c	2016-03-26 12:15:57.303729986 -0500
@@ -10,6 +10,7 @@
 #include <stdlib.h>
 #include <errno.h>
 #include <signal.h>
+#include <string.h>
 
 #include "criu.h"
 #include "rpc.pb-c.h"
