--- xfsprogs-5.4.0/db/crc.c
+++ xfsprogs-5.4.0/db/crc.c
@@ -5,6 +5,7 @@
  */
 
 #include "libxfs.h"
+#include "libxlog.h"
 #include "addr.h"
 #include "command.h"
 #include "type.h"
--- xfsprogs-5.4.0/db/fuzz.c
+++ xfsprogs-5.4.0/db/fuzz.c
@@ -5,6 +5,7 @@
  */
 
 #include "libxfs.h"
+#include "libxlog.h"
 #include <ctype.h>
 #include <time.h>
 #include "bit.h"
--- xfsprogs-5.4.0/db/info.c
+++ xfsprogs-5.4.0/db/info.c
@@ -4,6 +4,7 @@
  * Author: Darrick J. Wong <darrick.wong@oracle.com>
  */
 #include "libxfs.h"
+#include "libxlog.h"
 #include "command.h"
 #include "init.h"
 #include "output.h"
--- xfsprogs-5.4.0/db/init.c
+++ xfsprogs-5.4.0/db/init.c
@@ -27,7 +27,6 @@ static int		force;
 static struct xfs_mount	xmount;
 struct xfs_mount	*mp;
 static struct xlog	xlog;
-libxfs_init_t		x;
 xfs_agnumber_t		cur_agno = NULLAGNUMBER;
 
 static void
--- xfsprogs-5.4.0/db/init.h
+++ xfsprogs-5.4.0/db/init.h
@@ -9,5 +9,4 @@ extern int		blkbb;
 extern int		exitcode;
 extern int		expert_mode;
 extern xfs_mount_t	*mp;
-extern libxfs_init_t	x;
 extern xfs_agnumber_t	cur_agno;
--- xfsprogs-5.4.0/db/write.c
+++ xfsprogs-5.4.0/db/write.c
@@ -5,6 +5,7 @@
  */
 
 #include "libxfs.h"
+#include "libxlog.h"
 #include <ctype.h>
 #include <time.h>
 #include "bit.h"
--- xfsprogs-5.4.0/logprint/logprint.c
+++ xfsprogs-5.4.0/logprint/logprint.c
@@ -24,7 +24,6 @@ int	print_buffer;
 int	print_overwrite;
 int     print_no_data;
 int     print_no_print;
-int     print_exit = 1; /* -e is now default. specify -c to override */
 static int	print_operation = OP_PRINT;
 
 static void
@@ -133,6 +132,8 @@ main(int argc, char **argv)
 	textdomain(PACKAGE);
 	memset(&mount, 0, sizeof(mount));
 
+	print_exit = 1; /* -e is now default. specify -c to override */
+
 	progname = basename(argv[0]);
 	while ((c = getopt(argc, argv, "bC:cdefl:iqnors:tDVv")) != EOF) {
 		switch (c) {
--- xfsprogs-5.4.0/mdrestore/xfs_mdrestore.c
+++ xfsprogs-5.4.0/mdrestore/xfs_mdrestore.c
@@ -7,7 +7,6 @@
 #include "libxfs.h"
 #include "xfs_metadump.h"
 
-char 		*progname;
 static int	show_progress = 0;
 static int	show_info = 0;
 static int	progress_since_warning = 0;
