diff --git lm_sensors-3.3.5/prog/dump/isadump.c lm_sensors-fixed/prog/dump/isadump.c
index e031e47..3398612 100644
--- lm_sensors-3.3.5/prog/dump/isadump.c
+++ lm_sensors-fixed/prog/dump/isadump.c
@@ -33,17 +33,10 @@
 #include <stdlib.h>
 #include <unistd.h>
 #include <string.h>
+#include <sys/io.h>
 #include "util.h"
 #include "superio.h"
 
-
-/* To keep glibc2 happy */
-#if defined(__GLIBC__) && __GLIBC__ == 2 && __GLIBC_MINOR__ >= 0
-#include <sys/io.h>
-#else
-#include <asm/io.h>
-#endif
-
 #ifdef __powerpc__
 unsigned long isa_io_base = 0; /* XXX for now */
 #endif /* __powerpc__ */
diff --git lm_sensors-3.3.5/prog/dump/isaset.c lm_sensors-fixed/prog/dump/isaset.c
index 1d1bdad..a5443b9 100644
--- lm_sensors-3.3.5/prog/dump/isaset.c
+++ lm_sensors-fixed/prog/dump/isaset.c
@@ -30,15 +30,8 @@
 #include <stdlib.h>
 #include <unistd.h>
 #include <string.h>
-#include "util.h"
-
-
-/* To keep glibc2 happy */
-#if defined(__GLIBC__) && __GLIBC__ == 2 && __GLIBC_MINOR__ >= 0
 #include <sys/io.h>
-#else
-#include <asm/io.h>
-#endif
+#include "util.h"
 
 #ifdef __powerpc__
 unsigned long isa_io_base = 0; /* XXX for now */
diff --git lm_sensors-3.3.5/prog/dump/superio.c lm_sensors-fixed/prog/dump/superio.c
index 1af7358..a57cce6 100644
--- lm_sensors-3.3.5/prog/dump/superio.c
+++ lm_sensors-fixed/prog/dump/superio.c
@@ -20,13 +20,7 @@
 */
 
 #include <stdlib.h>
-
-#if defined(__GLIBC__) && __GLIBC__ == 2 && __GLIBC_MINOR__ >= 0
 #include <sys/io.h>
-#else
-#include <asm/io.h>
-#endif
-
 #include "superio.h"
 
 int superio_parse_key(unsigned char *key, const char *s)
diff --git lm_sensors-3.3.5/prog/dump/util.c lm_sensors-fixed/prog/dump/util.c
index 676c339..aab9fae 100644
--- lm_sensors-3.3.5/prog/dump/util.c
+++ lm_sensors-fixed/prog/dump/util.c
@@ -9,14 +9,8 @@
 */
 
 #include <stdio.h>
-#include "util.h"
-
-/* To keep glibc2 happy */
-#if defined(__GLIBC__) && __GLIBC__ == 2 && __GLIBC_MINOR__ >= 0
 #include <sys/io.h>
-#else
-#include <asm/io.h>
-#endif
+#include "util.h"
 
 /* Return 1 if we should continue, 0 if we should abort */
 int user_ack(int def)
