diff --git libcxx-3.9.0.src/include/stddef.h libcxx-3.9.0.src/include/stddef.h
index 8841bbe..a98f41d 100644
--- libcxx-3.9.0.src/include/stddef.h
+++ libcxx-3.9.0.src/include/stddef.h
@@ -53,7 +53,7 @@ using std::nullptr_t;
 }
 
 // Re-use the compiler's <stddef.h> max_align_t where possible.
-#if !defined(__CLANG_MAX_ALIGN_T_DEFINED) && !defined(_GCC_MAX_ALIGN_T)
+#if !defined(__CLANG_MAX_ALIGN_T_DEFINED) && !defined(_GCC_MAX_ALIGN_T) && !defined(__DEFINED_max_align_t)
 typedef long double max_align_t;
 #endif
 
