diff --git dwarves-1.13/CMakeLists.txt dwarves-1.13/CMakeLists.txt
--- dwarves-1.13/CMakeLists.txt
+++ dwarves-1.13/CMakeLists.txt
@@ -89,52 +89,52 @@ target_link_libraries(dwarves ${DWARF_LIBRARIES} ${ZLIB_LIBRARIES})
 set(dwarves_emit_LIB_SRCS dwarves_emit.c)
 add_library(dwarves_emit SHARED ${dwarves_emit_LIB_SRCS})
 set_target_properties(dwarves_emit PROPERTIES VERSION 1.0.0 SOVERSION 1)
-target_link_libraries(dwarves_emit dwarves)
+target_link_libraries(dwarves_emit argp dwarves)
 
 set(dwarves_reorganize_LIB_SRCS dwarves_reorganize.c)
 add_library(dwarves_reorganize SHARED ${dwarves_reorganize_LIB_SRCS})
 set_target_properties(dwarves_reorganize PROPERTIES VERSION 1.0.0 SOVERSION 1)
-target_link_libraries(dwarves_reorganize dwarves)
+target_link_libraries(dwarves_reorganize argp dwarves)
 
 set(codiff_SRCS codiff.c)
 add_executable(codiff ${codiff_SRCS})
-target_link_libraries(codiff dwarves)
+target_link_libraries(codiff argp dwarves)
 
 set(ctracer_SRCS ctracer.c)
 add_executable(ctracer ${ctracer_SRCS})
-target_link_libraries(ctracer dwarves dwarves_emit dwarves_reorganize ${ELF_LIBRARY})
+target_link_libraries(ctracer argp dwarves dwarves_emit dwarves_reorganize ${ELF_LIBRARY})
 
 set(dtagnames_SRCS dtagnames.c)
 add_executable(dtagnames ${dtagnames_SRCS})
-target_link_libraries(dtagnames dwarves)
+target_link_libraries(dtagnames argp dwarves)
 
 set(pahole_SRCS pahole.c)
 add_executable(pahole ${pahole_SRCS})
-target_link_libraries(pahole dwarves dwarves_reorganize)
+target_link_libraries(pahole argp dwarves dwarves_reorganize)
 
 set(pdwtags_SRCS pdwtags.c)
 add_executable(pdwtags ${pdwtags_SRCS})
-target_link_libraries(pdwtags dwarves)
+target_link_libraries(pdwtags argp dwarves)
 
 set(pglobal_SRCS pglobal.c)
 add_executable(pglobal ${pglobal_SRCS})
-target_link_libraries(pglobal dwarves)
+target_link_libraries(pglobal argp dwarves)
 
 set(pfunct_SRCS pfunct.c)
 add_executable(pfunct ${pfunct_SRCS})
-target_link_libraries(pfunct dwarves dwarves_emit ${ELF_LIBRARY})
+target_link_libraries(pfunct argp dwarves dwarves_emit ${ELF_LIBRARY})
 
 set(prefcnt_SRCS prefcnt.c)
 add_executable(prefcnt ${prefcnt_SRCS})
-target_link_libraries(prefcnt dwarves)
+target_link_libraries(prefcnt argp dwarves)
 
 set(scncopy_SRCS scncopy.c elfcreator.c)
 add_executable(scncopy ${scncopy_SRCS})
-target_link_libraries(scncopy dwarves ${ELF_LIBRARY})
+target_link_libraries(scncopy argp dwarves ${ELF_LIBRARY})
 
 set(syscse_SRCS syscse.c)
 add_executable(syscse ${syscse_SRCS})
-target_link_libraries(syscse dwarves)
+target_link_libraries(syscse argp dwarves)
 
 install(TARGETS codiff ctracer dtagnames pahole pdwtags
 		pfunct pglobal prefcnt scncopy syscse RUNTIME DESTINATION
