--- psutil-5.6.7/psutil/_psutil_linux.c
+++ psutil-5.6.7/psutil/_psutil_linux.c
@@ -45,7 +45,7 @@ static const int NCPUS_START = sizeof(unsigned long) * CHAR_BIT;
 // Linux >= 2.6.36 (supposedly) and glibc >= 13
 #define PSUTIL_HAVE_PRLIMIT \
     (LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 36)) && \
-    (__GLIBC__ >= 2 && __GLIBC_MINOR__ >= 13) && \
+    (!defined(__GLIBC__) || (__GLIBC__ >= 2 && __GLIBC_MINOR__ >= 13)) && \
     defined(__NR_prlimit64)
 
 #if PSUTIL_HAVE_PRLIMIT
