--- a/psutil/tests/test_linux.py
+++ b/psutil/tests/test_linux.py
@@ -244,7 +244,8 @@ class TestSystemVirtualMemory(PsutilTestCase):
         # self.assertEqual(free_value, psutil_value)
         vmstat_value = vmstat('total memory') * 1024
         psutil_value = psutil.virtual_memory().total
-        self.assertAlmostEqual(vmstat_value, psutil_value)
+        self.assertAlmostEqual(
+            vmstat_value, psutil_value, delta=MEMORY_TOLERANCE)
 
     @retry_on_failure()
     def test_used(self):
