--- psutil-5.9.0/psutil/_pslinux.py
+++ psutil-5.9.0/psutil/_pslinux.py
@@ -728,6 +728,9 @@ def _cpu_get_cpuinfo_freq():
         for line in f:
             if line.lower().startswith(b'cpu mhz'):
                 ret.append(float(line.split(b':', 1)[1]))
+            elif line.lower().startswith(b'clock\t'):
+                key, value = line.split(b'\t:', 1)
+                ret.append(float(value[:-4]))
     return ret
 
 
