--- a/tests.c
+++ b/tests.c
@@ -22,6 +22,16 @@
 #include "sizes.h"
 #include "memtester.h"
 
+static union {
+    unsigned char bytes[UL_LEN/8];
+    ul val;
+} mword8;
+
+static union {
+    unsigned short u16s[UL_LEN/16];
+    ul val;
+} mword16;
+
 char progress[] = "-\\|/";
 #define PROGRESSLEN 4
 #define PROGRESSOFTEN 2500
--- a/types.h
+++ b/types.h
@@ -24,13 +24,3 @@ struct test {
     char *name;
     int (*fp)();
 };
-
-union {
-    unsigned char bytes[UL_LEN/8];
-    ul val;
-} mword8;
-
-union {
-    unsigned short u16s[UL_LEN/16];
-    ul val;
-} mword16;
