diff -ur util-linux-2.26.2/login-utils/lslogins.c util-linux-2.26.2-fixed/login-utils/lslogins.c
--- util-linux-2.26.2/login-utils/lslogins.c	2015-04-29 07:52:21.984364178 +0000
+++ util-linux-2.26.2-fixed/login-utils/lslogins.c	2016-01-30 21:59:44.465253096 +0000
@@ -23,6 +23,7 @@
 #include <stdlib.h>
 #include <unistd.h>
 #include <getopt.h>
+#include <sys/time.h>
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <sys/syslog.h>
