diff --git firefox-52.0.1/extensions/spellcheck/hunspell/glue/moz.build
index 1e1d971..1afe16b 100644
--- firefox-52.0.1/extensions/spellcheck/hunspell/glue/moz.build.orig
+++ firefox-52.0.1/extensions/spellcheck/hunspell/glue/moz.build
@@ -33,7 +33,3 @@ EXPORTS.mozilla += [
      'RemoteSpellCheckEngineChild.h',
      'RemoteSpellCheckEngineParent.h',
 ]
-
-# This variable is referenced in configure.in.  Make sure to change that file
-# too if you need to change this variable.
-DEFINES['HUNSPELL_STATIC'] = True
