From f2b72df6a4afccde81716b3026e0da637fbc4c8a Mon Sep 17 00:00:00 2001
From: Samuel Holland <samuel@sholland.org>
Date: Sun, 7 Jul 2019 02:20:51 -0500
Subject: [PATCH 2/9] Link libssp_nonshared when SSP is enabled

---
 gcc/gcc.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/gcc/gcc.c b/gcc/gcc.c
index 81e47efa2e00..ec1109988687 100644
--- a/gcc/gcc.c
+++ b/gcc/gcc.c
@@ -979,8 +979,7 @@ proper position among the other output files.  */
 
 #ifndef LINK_SSP_SPEC
 #ifdef TARGET_LIBC_PROVIDES_SSP
-#define LINK_SSP_SPEC "%{fstack-protector|fstack-protector-all" \
-		       "|fstack-protector-strong|fstack-protector-explicit:}"
+#define LINK_SSP_SPEC "%{!fno-stack-protector:-lssp_nonshared}"
 #else
 #define LINK_SSP_SPEC "%{fstack-protector|fstack-protector-all" \
 		       "|fstack-protector-strong|fstack-protector-explicit" \
-- 
2.31.1

