From b09fe5bbdb19e327e11a903d3709d622592f25df Mon Sep 17 00:00:00 2001
From: Samuel Holland <samuel@sholland.org>
Date: Sun, 7 Jul 2019 02:20:51 -0500
Subject: [PATCH 02/10] Link libssp_nonshared when SSP is enabled

---
 gcc/gcc.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/gcc/gcc.c b/gcc/gcc.c
index eec0886e2853..0f2ec942aacc 100644
--- a/gcc/gcc.c
+++ b/gcc/gcc.c
@@ -979,8 +979,7 @@ proper position among the other output files.  */
 
 #ifndef LINK_SSP_SPEC
 #ifdef TARGET_LIBC_PROVIDES_SSP
-#define LINK_SSP_SPEC "%{fstack-protector|fstack-protector-all" \
-		       "|fstack-protector-strong|fstack-protector-explicit:}"
+#define LINK_SSP_SPEC "%{!fno-stack-protector:-lssp_nonshared}"
 #else
 #define LINK_SSP_SPEC "%{fstack-protector|fstack-protector-all" \
 		       "|fstack-protector-strong|fstack-protector-explicit" \
-- 
2.33.1

