diff --git a/libgcc/config/rs6000/t-linux b/libgcc/config/rs6000/t-linux
index 500210ddaf20..9d573adc8ccc 100644
--- a/libgcc/config/rs6000/t-linux
+++ b/libgcc/config/rs6000/t-linux
@@ -1,7 +1,5 @@
 SHLIB_MAPFILES += $(srcdir)/config/rs6000/libgcc-glibc.ver
 
-HOST_LIBGCC2_CFLAGS += -mlong-double-128
-
 # This is a way of selecting -mcmodel=small for ppc64, which gives
 # smaller and faster libgcc code.  Directly specifying -mcmodel=small
 # would need to take into account targets for which -mcmodel is invalid.
diff --git a/libgcc/config/rs6000/t-ppccomm b/libgcc/config/rs6000/t-ppccomm
index fb812d3801fe..6a82d4bcd35e 100644
--- a/libgcc/config/rs6000/t-ppccomm
+++ b/libgcc/config/rs6000/t-ppccomm
@@ -1,4 +1,4 @@
-LIB2ADD += $(srcdir)/config/rs6000/ibm-ldouble.c \
+LIB2ADD += \
 	   $(srcdir)/config/rs6000/tramp.S
 
 # This can't end up in shared libgcc
