diff --git iperf-3.0.12/src/cjson.h iperf-fixed/src/cjson.h
index fb8cd3b..69cd272 100644
--- iperf-3.0.12/src/cjson.h
+++ iperf-fixed/src/cjson.h
@@ -23,6 +23,8 @@
 #ifndef cJSON__h
 #define cJSON__h
 
+#include <stdint.h>
+
 #ifdef __cplusplus
 extern "C"
 {
diff --git iperf-3.1.3/src/iperf_util.h iperf-fixed/src/iperf_util.h
index e422a11..2f5f7db 100644
--- iperf-3.1.3/src/iperf_util.h
+++ iperf-fixed/src/iperf_util.h
@@ -28,6 +28,7 @@
 #define __IPERF_UTIL_H
 
 #include "cjson.h"
+#include <stdint.h>
 #include <sys/select.h>
 
 void make_cookie(char *);
diff --git iperf-3.0.12/src/timer.h iperf-fixed/src/timer.h
index 0f9c5eb..9b22b7f 100644
--- iperf-3.0.12/src/timer.h
+++ iperf-fixed/src/timer.h
@@ -30,6 +30,7 @@
 #ifndef __TIMER_H
 #define __TIMER_H
 
+#include <stdint.h>
 #include <sys/time.h>
 
 /* TimerClientData is an opaque value that tags along with a timer.  The
