From 9c46fc2657dfae756d17a8a25ab8cd6c124f1f7b Mon Sep 17 00:00:00 2001
From: Prajith <me@prajith.in>
Date: Thu, 2 Jan 2020 11:51:23 +0530
Subject: [PATCH] Update Test t/getdate.t

---
 t/getdate.t | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/t/getdate.t b/t/getdate.t
index 31b577b..82c5850 100644
--- a/t/getdate.t
+++ b/t/getdate.t
@@ -156,7 +156,7 @@ Jul 22 10:00:00 UTC 2002	     ;1027332000
 !;
 
 require Time::Local;
-my $offset = Time::Local::timegm(0,0,0,1,0,70);
+my $offset = Time::Local::timegm(0,0,0,1,0,1970);
 
 @data = split(/\n/, $data);
 
