diff --git ddcutil-0.8.4/src/util/debug_util.c
index 52e55f7..824a2d6 100644
--- ddcutil-0.8.4/src/util/debug_util.c.orig
+++ ddcutil-0.8.4/src/util/debug_util.c
@@ -26,6 +26,7 @@
  */
 
 /** \cond */
+#ifdef __GLIBC__
 #include <execinfo.h>
 #include <stdbool.h>
 #include <stdio.h>
@@ -120,3 +121,8 @@ void show_backtrace(int stack_adjust)
       free(strings);
    }
 }
+#else
+void show_backtrace(int stack_adjust)
+{
+}
+#endif
