diff --git sandbox-2.13/libsbutil/get_sandbox_lib.c
index b64a5ac..a8d9a00 100644
--- sandbox-2.13/libsbutil/get_sandbox_lib.c.orig
+++ sandbox-2.13/libsbutil/get_sandbox_lib.c
@@ -21,14 +21,5 @@
  */
 void get_sandbox_lib(char *path)
 {
-	save_errno();
-	strcpy(path, LIB_NAME);
-	if (strncmp("/usr/lib", LIBSANDBOX_PATH, 8)) {
-		void *hndl = dlopen(path, RTLD_LAZY);
-		if (!hndl)
-			snprintf(path, SB_PATH_MAX, "%s/%s", LIBSANDBOX_PATH, LIB_NAME);
-		else
-			dlclose(hndl);
-	}
-	restore_errno();
+	strcpy(path, LIBSANDBOX_PATH "/" LIB_NAME);
 }
