--- nullmailer-2.0/test/functions.in
+++ nullmailer-2.0/test/functions.in
@@ -16,7 +16,7 @@ ln -s $builddir/src $tmpdir/bin
 ln -s $builddir/protocols/* $tmpdir/protocols/
 
 exit_cleanup() {
-    svc -dx $tmpdir/service/* > /dev/null 2>&1 || :
+    s6-svc -dx $tmpdir/service/* > /dev/null 2>&1 || :
     # Remove all temporary files on success
     if [ $? -eq 0 ]
     then
@@ -38,11 +38,11 @@ start() {
 	echo exec "$@"
     } >$tmpdir/service/$name/run
     chmod +x $tmpdir/service/$name/run
-    supervise $tmpdir/service/$name </dev/null >$tmpdir/service/${name}-log 2>&1 &
+    s6-supervise $tmpdir/service/$name </dev/null >$tmpdir/service/${name}-log 2>&1 &
 }
 stop() {
     for name in $*; do
-	svc -dx $tmpdir/service/$name >/dev/null 2>&1
+	s6-svc -dx $tmpdir/service/$name >/dev/null 2>&1
     done
 }
 
--- nullmailer-2.0/test/tests/send
+++ nullmailer-2.0/test/tests/send
@@ -33,7 +33,7 @@ EOF
 send_message() {
   echo 127.0.0.1 dummy $@ >$SYSCONFDIR/remotes
   make_message
-  svc -a $tmpdir/service/send
+  s6-svc -a $tmpdir/service/send
   sleep 2
   not test -e $QUEUEDIR/queue/$msgid
 }
@@ -81,3 +81,5 @@ grep -qx "^Starting delivery: host: 127.0.0.1 protocol: dummy file: $msgid" $log
 grep -qx 'From: <me@example.com> to: <me@example.net>' $log
 grep -qx 'Sending failed: Unspecified temporary error' $log
 grep -qx "Message-Id: <$msgid>" $log
+
+stop send
--- nullmailer-2.0/test/tests/protocols
+++ nullmailer-2.0/test/tests/protocols
@@ -15,7 +15,7 @@ EOF
 
 for p in smtp qmqp
 do
-	start server "tcpserver -1 ::0 0 sh $srcdir/test/accept-$p.sh"
+	start server "s6-tcpserver -1 :: 0 sh $srcdir/test/accept-$p.sh"
 	sleep 1
 	port=$( head -n 1 $tmpdir/service/server-log )
 
@@ -39,15 +39,6 @@ do
 
 	stop server
 
-	echo "Testing host not found error with $p."
-	error 2 protocol $p --host=this.host.can.not.exist 3<testmail
-
-	echo "Testing connection refused error with $p."
-	error 7 protocol $p -p $port --host=localhost 3<testmail
-
-	echo "Testing connection refused error with $p (IPv6)."
-	error 7 protocol $p -p $port --host=::1 3<testmail
-	
 	echo "Testing usage error with $p (zero args)."
 	error 1 protocol $p 3<testmail
 	
@@ -60,7 +54,7 @@ do
 	echo "Testing usage error with $p (invalid integer)."
 	error 1 protocol $p -p foo --host=localhost 3<testmail
 
-	start server tcpserver -1 0 0 date
+	start server s6-tcpserver -1 :: 0 date
 	sleep 1
 	port=$( head -n 1 $tmpdir/service/server-log )
 	echo "Testing protocol failure with $p."
