--- jsoncpp-1.9.2/src/lib_json/json_writer.cpp
+++ jsoncpp-1.9.2/src/lib_json/json_writer.cpp
@@ -179,7 +179,7 @@ static bool isAnyCharRequiredQuoting(char const* s, size_t n) {
   char const* const end = s + n;
   for (char const* cur = s; cur < end; ++cur) {
     if (*cur == '\\' || *cur == '\"' || *cur < ' ' ||
-        static_cast<unsigned char>(*cur) < 0x80)
+        static_cast<unsigned char>(*cur) >= 0x80)
       return true;
   }
   return false;
