--- vis-v0.5/test/core/ccan/tap/tap.c
+++ vis-v0.5/test/core/ccan/tap/tap.c
@@ -32,6 +32,8 @@
 
 #include "tap.h"
 
+void (*tap_fail_callback)(void);
+
 static int no_plan = 0;
 static int skip_all = 0;
 static int have_plan = 0;
--- vis-v0.5/test/core/ccan/tap/tap.h
+++ vis-v0.5/test/core/ccan/tap/tap.h
@@ -246,6 +246,6 @@ void plan_skip_all(const char *reason);
  *
  * This can be used to ease debugging, or exit on the first failure.
  */
-void (*tap_fail_callback)(void);
+extern void (*tap_fail_callback)(void);
 
 #endif /* CCAN_TAP_H */
