- strerror() difference
- flushing
--- gawk-4.1.4/test/testext.ok
+++ gawk-4.1.4/test/testext.ok
@@ -21,7 +21,7 @@ var_test: sym_update of ARGC failed - correctly
 var_test: sym_update("testvar") succeeded
 var_test() returned 1, test_var = 42
 
-test_errno() returned 1, ERRNO = No child processes
+test_errno() returned 1, ERRNO = No child process
 
 length of test_array is 10, should be 10
 test_array_size: incoming size is 10
--- gawk-4.2.1/test/clos1way6.ok
+++ gawk-4.2.1/test/clos1way6.ok
@@ -1,3 +1,3 @@
 gawk: clos1way6.awk:5: warning: fflush: cannot flush: two-way pipe `cat - 1>&2; sleep 2' has closed write end
-test1
 Bad file descriptor
+test1
