--- freerdp-9999/winpr/libwinpr/crt/test/TestUnicodeConversion.c
+++ freerdp-9999/winpr/libwinpr/crt/test/TestUnicodeConversion.c
@@ -402,6 +402,7 @@ static BOOL test_ConvertToUnicode_wrapper(void)
 
 	/* Test static string buffers of differing sizes */
 	{
+# if 0
 		char name[] = "someteststring";
 		const WCHAR cmp[] = { L's', L'o', L'm', L'e', L't', L'e', L's', L't',
 			                  L's', L't', L'r', L'i', L'n', L'g', 0 };
@@ -423,6 +424,7 @@ static BOOL test_ConvertToUnicode_wrapper(void)
 		free(aname);
 		if (ii != 0)
 			goto fail;
+#endif
 	}
 
 	/* Test unterminated unicode string:
