diff --git freerdp-9999/libfreerdp/crypto/crypto.c
index dadc388..ef66ecf 100644
--- freerdp-9999/libfreerdp/crypto/crypto.c.orig
+++ freerdp-9999/libfreerdp/crypto/crypto.c
@@ -388,7 +388,7 @@ BOOL x509_verify_certificate(CryptoCert cert, char* certificate_store_path)
 	if (cert_ctx == NULL)
 		goto end;
 
-#if OPENSSL_VERSION_NUMBER < 0x10100000L
+#if OPENSSL_VERSION_NUMBER < 0x10100000L || defined(LIBRESSL_VERSION_NUMBER)
 	OpenSSL_add_all_algorithms();
 #else
 	OPENSSL_init_crypto(OPENSSL_INIT_ADD_ALL_CIPHERS \
diff --git freerdp-9999/winpr/libwinpr/utils/ssl.c
index b674e21..eb924f1 100644
--- freerdp-9999/winpr/libwinpr/utils/ssl.c.orig
+++ freerdp-9999/winpr/libwinpr/utils/ssl.c
@@ -255,7 +255,7 @@ static BOOL CALLBACK _winpr_openssl_initialize(PINIT_ONCE once, PVOID param, PVO
 	}
 #endif
 	/* SSL_load_error_strings() is void */
-#if (OPENSSL_VERSION_NUMBER < 0x10100000L)
+#if (OPENSSL_VERSION_NUMBER < 0x10100000L) || defined(LIBRESSL_VERSION_NUMBER)
 	SSL_load_error_strings();
 	/* SSL_library_init() always returns "1" */
 	SSL_library_init();
diff --git freerdp-9999/winpr/tools/makecert/makecert.c
index 2fdb716..87c90d4 100644
--- freerdp-9999/winpr/tools/makecert/makecert.c.orig
+++ freerdp-9999/winpr/tools/makecert/makecert.c
@@ -620,7 +620,7 @@ int makecert_context_output_certificate_file(MAKECERT_CONTEXT* context, char* pa
 				printf("Using default export password \"password\"\n");
 			}
 
-#if OPENSSL_VERSION_NUMBER < 0x10100000L
+#if OPENSSL_VERSION_NUMBER < 0x10100000L || defined(LIBRESSL_VERSION_NUMBER)
 			OpenSSL_add_all_algorithms();
 			OpenSSL_add_all_ciphers();
 			OpenSSL_add_all_digests();
@@ -1008,7 +1008,7 @@ int makecert_context_process(MAKECERT_CONTEXT* context, int argc, char** argv)
 		key_length = atoi(arg->Value);
 	}
 
-#if (OPENSSL_VERSION_NUMBER < 0x10100000L)
+#if (OPENSSL_VERSION_NUMBER < 0x10100000L) || defined(LIBRESSL_VERSION_NUMBER)
 	context->rsa = RSA_generate_key(key_length, RSA_F4, NULL, NULL);
 #else
 	{
@@ -1043,7 +1043,7 @@ int makecert_context_process(MAKECERT_CONTEXT* context, int argc, char** argv)
 	{
 		ASN1_TIME* before;
 		ASN1_TIME* after;
-#if (OPENSSL_VERSION_NUMBER < 0x10100000L)
+#if (OPENSSL_VERSION_NUMBER < 0x10100000L) || defined(LIBRESSL_VERSION_NUMBER)
 		before = X509_get_notBefore(context->x509);
 		after = X509_get_notAfter(context->x509);
 #else
