--- skiboot-6.5/ccan/endian/endian.h
+++ skiboot-6.5/ccan/endian/endian.h
@@ -108,6 +108,7 @@ static inline uint64_t bswap_64(uint64_t val)
 #define	__BIG_ENDIAN	4321
 
 /* Sanity check the defines.  We don't handle weird endianness. */
+#undef __BYTE_ORDER
 #if !HAVE_LITTLE_ENDIAN && !HAVE_BIG_ENDIAN
 #error "Unknown endian"
 #elif HAVE_LITTLE_ENDIAN && HAVE_BIG_ENDIAN
--- skiboot-6.5/external/opal-prd/thunk.S
+++ skiboot-6.5/external/opal-prd/thunk.S
@@ -1,7 +1,7 @@
 // SPDX-License-Identifier: Apache-2.0
 /* Copyright 2015-2017 IBM Corp */
 	
-#include <endian.h>
+#include <bits/endian.h>
 #include <asm/unistd.h>
 
 #ifndef __NR_switch_endian
--- skiboot-6.5/include/inttypes.h
+++ skiboot-6.5/include/inttypes.h
@@ -10,17 +10,17 @@
 #ifndef __SKIBOOT_INTTYPES_H
 #define __SKIBOOT_INTTYPES_H
 
-#include <stdint.h>
+#include <limits.h>
 
-#ifndef __WORDSIZE
-/* If we don't have __WORDSIZE it means we're *certainly* building skiboot
+#ifndef LONG_BIT
+/* If we don't have LONG_BIT it means we're *certainly* building skiboot
  * which will *ALWAYS* have a word size of 32bits.
  * (unless someone goes and ports skiboot to something that isn't powerpc)
  */
-#define __WORDSIZE 32
+#define LONG_BIT 32
 #endif
 
-#if __WORDSIZE == 64
+#if LONG_BIT == 64
 #define PRIu64 "lu"
 #define PRIx64 "lx"
 #else
