diff --git source/i18n/digitlst.cpp source-fixed/i18n/digitlst.cpp
index a3b3406..80eb098 100644
--- source/i18n/digitlst.cpp
+++ source-fixed/i18n/digitlst.cpp
@@ -61,11 +61,7 @@
 #endif
 
 #if U_USE_STRTOD_L && !U_PLATFORM_USES_ONLY_WIN32_API
-# if U_PLATFORM == U_PF_CYGWIN
-#   include <locale.h>
-# else
-#   include <xlocale.h>
-# endif
+# include <locale.h>
 #endif
 
 // ***************************************************************************
