--- gnupg-2.2.8/common/iobuf.h
+++ gnupg-2.2.8/common/iobuf.h
@@ -250,7 +250,7 @@ struct iobuf_struct
 };
 
 #ifndef EXTERN_UNLESS_MAIN_MODULE
-#if defined (__riscos__) && !defined (INCLUDED_BY_MAIN_MODULE)
+#if !defined (INCLUDED_BY_MAIN_MODULE)
 #define EXTERN_UNLESS_MAIN_MODULE extern
 #else
 #define EXTERN_UNLESS_MAIN_MODULE
--- gnupg-2.2.8/common/t-iobuf.c
+++ gnupg-2.2.8/common/t-iobuf.c
@@ -4,6 +4,7 @@
 #include <assert.h>
 #include <stdlib.h>
 
+#define INCLUDED_BY_MAIN_MODULE 1
 #include "iobuf.h"
 #include "stringhelp.h"
 
--- gnupg-2.2.8/g10/options.h
+++ gnupg-2.2.8/g10/options.h
@@ -31,8 +31,7 @@
 #include "../common/compliance.h"
 
 #ifndef EXTERN_UNLESS_MAIN_MODULE
-/* Norcraft can't cope with common symbols */
-#if defined (__riscos__) && !defined (INCLUDED_BY_MAIN_MODULE)
+#if !defined (INCLUDED_BY_MAIN_MODULE)
 #define EXTERN_UNLESS_MAIN_MODULE extern
 #else
 #define EXTERN_UNLESS_MAIN_MODULE
--- gnupg-2.2.8/sm/gpgsm.c
+++ gnupg-2.2.8/sm/gpgsm.c
@@ -28,6 +28,7 @@
 #include <fcntl.h>
 /*#include <mcheck.h>*/
 
+#define INCLUDED_BY_MAIN_MODULE 1
 #include "gpgsm.h"
 #include <gcrypt.h>
 #include <assuan.h> /* malloc hooks */
--- gnupg-2.2.8/sm/gpgsm.h
+++ gnupg-2.2.8/sm/gpgsm.h
@@ -51,7 +51,16 @@ struct keyserver_spec
 };
 
 
+#ifndef EXTERN_UNLESS_MAIN_MODULE
+#if !defined (INCLUDED_BY_MAIN_MODULE)
+#define EXTERN_UNLESS_MAIN_MODULE extern
+#else
+#define EXTERN_UNLESS_MAIN_MODULE
+#endif
+#endif
+
 /* A large struct named "opt" to keep global flags. */
+EXTERN_UNLESS_MAIN_MODULE
 struct
 {
   unsigned int debug; /* debug flags (DBG_foo_VALUE) */
--- gnupg-2.2.8/agent/agent.h
+++ gnupg-2.2.8/agent/agent.h
@@ -54,7 +54,16 @@
 #define MAX_PASSPHRASE_LEN 255
 
 
-/* A large struct name "opt" to keep global flags */
+#ifndef EXTERN_UNLESS_MAIN_MODULE
+#if !defined (INCLUDED_BY_MAIN_MODULE)
+#define EXTERN_UNLESS_MAIN_MODULE extern
+#else
+#define EXTERN_UNLESS_MAIN_MODULE
+#endif
+#endif
+
+/* A large struct named "opt" to keep global flags. */
+EXTERN_UNLESS_MAIN_MODULE
 struct
 {
   unsigned int debug;  /* Debug flags (DBG_foo_VALUE) */
--- gnupg-2.2.8/agent/gpg-agent.c
+++ gnupg-2.2.8/agent/gpg-agent.c
@@ -50,6 +50,7 @@
 #include <npth.h>
 
 #define GNUPG_COMMON_NEED_AFLOCAL
+#define INCLUDED_BY_MAIN_MODULE 1
 #include "agent.h"
 #include <assuan.h> /* Malloc hooks  and socket wrappers. */
 
--- gnupg-2.2.8/agent/protect-tool.c
+++ gnupg-2.2.8/agent/protect-tool.c
@@ -38,6 +38,7 @@
 #include <fcntl.h> /* for setmode() */
 #endif
 
+#define INCLUDED_BY_MAIN_MODULE 1
 #include "agent.h"
 #include "../common/i18n.h"
 #include "../common/get-passphrase.h"
--- gnupg-2.2.8/agent/preset-passphrase.c
+++ gnupg-2.2.8/agent/preset-passphrase.c
@@ -44,6 +44,7 @@
 # include <windows.h>  /* To initialize the sockets.  fixme */
 #endif
 
+#define INCLUDED_BY_MAIN_MODULE 1
 #include "agent.h"
 #include "../common/simple-pwquery.h"
 #include "../common/i18n.h"
--- gnupg-2.2.8/agent/t-protect.c
+++ gnupg-2.2.8/agent/t-protect.c
@@ -23,6 +23,7 @@
 #include <stdlib.h>
 #include <assert.h>
 
+#define INCLUDED_BY_MAIN_MODULE 1
 #include "agent.h"
 
 
--- gnupg-2.2.8/scd/scdaemon.h
+++ gnupg-2.2.8/scd/scdaemon.h
@@ -43,7 +43,16 @@
 
 
 
-/* A large struct name "opt" to keep global flags. */
+#ifndef EXTERN_UNLESS_MAIN_MODULE
+#if !defined (INCLUDED_BY_MAIN_MODULE)
+#define EXTERN_UNLESS_MAIN_MODULE extern
+#else
+#define EXTERN_UNLESS_MAIN_MODULE
+#endif
+#endif
+
+/* A large struct named "opt" to keep global flags. */
+EXTERN_UNLESS_MAIN_MODULE
 struct
 {
   unsigned int debug; /* Debug flags (DBG_foo_VALUE). */
--- gnupg-2.2.8/scd/scdaemon.c
+++ gnupg-2.2.8/scd/scdaemon.c
@@ -38,6 +38,7 @@
 #include <npth.h>
 
 #define GNUPG_COMMON_NEED_AFLOCAL
+#define INCLUDED_BY_MAIN_MODULE 1
 #include "scdaemon.h"
 #include <ksba.h>
 #include <gcrypt.h>
--- gnupg-2.2.8/dirmngr/dirmngr.h
+++ gnupg-2.2.8/dirmngr/dirmngr.h
@@ -74,7 +74,16 @@ struct fingerprint_list_s
 };
 
 
-/* A large struct named "opt" to keep global flags.  */
+#ifndef EXTERN_UNLESS_MAIN_MODULE
+#if !defined (INCLUDED_BY_MAIN_MODULE)
+#define EXTERN_UNLESS_MAIN_MODULE extern
+#else
+#define EXTERN_UNLESS_MAIN_MODULE
+#endif
+#endif
+
+/* A large struct named "opt" to keep global flags. */
+EXTERN_UNLESS_MAIN_MODULE
 struct
 {
   unsigned int debug; /* debug flags (DBG_foo_VALUE) */
--- gnupg-2.2.8/dirmngr/dirmngr.c
+++ gnupg-2.2.8/dirmngr/dirmngr.c
@@ -56,6 +56,7 @@
 
 
 #define GNUPG_COMMON_NEED_AFLOCAL
+#define INCLUDED_BY_MAIN_MODULE 1
 #include "dirmngr.h"
 
 #include <assuan.h>
--- gnupg-2.2.8/tools/gpgtar.h
+++ gnupg-2.2.8/tools/gpgtar.h
@@ -23,7 +23,16 @@
 #include "../common/util.h"
 #include "../common/strlist.h"
 
-/* We keep all global options in the structure OPT.  */
+#ifndef EXTERN_UNLESS_MAIN_MODULE
+#if !defined (INCLUDED_BY_MAIN_MODULE)
+#define EXTERN_UNLESS_MAIN_MODULE extern
+#else
+#define EXTERN_UNLESS_MAIN_MODULE
+#endif
+#endif
+
+/* A large struct named "opt" to keep global flags. */
+EXTERN_UNLESS_MAIN_MODULE
 struct
 {
   int verbose;
--- gnupg-2.2.8/tools/gpgtar.c
+++ gnupg-2.2.8/tools/gpgtar.c
@@ -41,6 +41,7 @@
 #include "../common/init.h"
 #include "../common/strlist.h"
 
+#define INCLUDED_BY_MAIN_MODULE 1
 #include "gpgtar.h"
 
 
--- gnupg-2.2.8/tools/gpg-wks.h
+++ gnupg-2.2.8/tools/gpg-wks.h
@@ -29,7 +29,16 @@
 #define WKS_DRAFT_VERSION 3
 
 
-/* We keep all global options in the structure OPT.  */
+#ifndef EXTERN_UNLESS_MAIN_MODULE
+#if !defined (INCLUDED_BY_MAIN_MODULE)
+#define EXTERN_UNLESS_MAIN_MODULE extern
+#else
+#define EXTERN_UNLESS_MAIN_MODULE
+#endif
+#endif
+
+/* A large struct named "opt" to keep global flags. */
+EXTERN_UNLESS_MAIN_MODULE
 struct
 {
   int verbose;
--- gnupg-2.2.8/tools/gpg-wks-client.c
+++ gnupg-2.2.8/tools/gpg-wks-client.c
@@ -37,6 +37,7 @@
 #include "call-dirmngr.h"
 #include "mime-maker.h"
 #include "send-mail.h"
+#define INCLUDED_BY_MAIN_MODULE 1
 #include "gpg-wks.h"
 
 
--- gnupg-2.2.8/tools/gpgconf.h
+++ gnupg-2.2.8/tools/gpgconf.h
@@ -22,7 +22,16 @@
 
 #include "../common/util.h"
 
-/* We keep all global options in the structure OPT.  */
+#ifndef EXTERN_UNLESS_MAIN_MODULE
+#if !defined (INCLUDED_BY_MAIN_MODULE)
+#define EXTERN_UNLESS_MAIN_MODULE extern
+#else
+#define EXTERN_UNLESS_MAIN_MODULE
+#endif
+#endif
+
+/* A large struct named "opt" to keep global flags. */
+EXTERN_UNLESS_MAIN_MODULE
 struct
 {
   int verbose;		/* Verbosity level.  */
--- gnupg-2.2.8/tools/gpgconf.c
+++ gnupg-2.2.8/tools/gpgconf.c
@@ -25,6 +25,7 @@
 #include <string.h>
 #include <unistd.h>
 
+#define INCLUDED_BY_MAIN_MODULE 1
 #include "gpgconf.h"
 #include "../common/i18n.h"
 #include "../common/sysutils.h"
