diff --git gnupg-2.2.8/common/iobuf.h.orig gnupg-2.2.8/common/iobuf.h
index 22e02da..07c968d 100644
--- gnupg-2.2.8/common/iobuf.h.orig
+++ gnupg-2.2.8/common/iobuf.h
@@ -250,7 +250,7 @@ struct iobuf_struct
 };
 
 #ifndef EXTERN_UNLESS_MAIN_MODULE
-#if defined (__riscos__) && !defined (INCLUDED_BY_MAIN_MODULE)
+#if !defined (INCLUDED_BY_MAIN_MODULE)
 #define EXTERN_UNLESS_MAIN_MODULE extern
 #else
 #define EXTERN_UNLESS_MAIN_MODULE
diff --git gnupg-2.2.8/common/t-iobuf.c.orig gnupg-2.2.8/common/t-iobuf.c
index bdeab99..2d07a29 100644
--- gnupg-2.2.8/common/t-iobuf.c.orig
+++ gnupg-2.2.8/common/t-iobuf.c
@@ -4,6 +4,7 @@
 #include <assert.h>
 #include <stdlib.h>
 
+#define INCLUDED_BY_MAIN_MODULE 1
 #include "iobuf.h"
 #include "stringhelp.h"
 
diff --git gnupg-2.2.8/g10/options.h.orig gnupg-2.2.8/g10/options.h
index 6c67265..928f3b6 100644
--- gnupg-2.2.8/g10/options.h.orig
+++ gnupg-2.2.8/g10/options.h
@@ -31,8 +31,7 @@
 #include "../common/compliance.h"
 
 #ifndef EXTERN_UNLESS_MAIN_MODULE
-/* Norcraft can't cope with common symbols */
-#if defined (__riscos__) && !defined (INCLUDED_BY_MAIN_MODULE)
+#if !defined (INCLUDED_BY_MAIN_MODULE)
 #define EXTERN_UNLESS_MAIN_MODULE extern
 #else
 #define EXTERN_UNLESS_MAIN_MODULE
diff --git gnupg-2.2.8/sm/gpgsm.c.orig gnupg-2.2.8/sm/gpgsm.c
index da1783d..61e5a6b 100644
--- gnupg-2.2.8/sm/gpgsm.c.orig
+++ gnupg-2.2.8/sm/gpgsm.c
@@ -28,6 +28,7 @@
 #include <fcntl.h>
 /*#include <mcheck.h>*/
 
+#define INCLUDED_BY_MAIN_MODULE 1
 #include "gpgsm.h"
 #include <gcrypt.h>
 #include <assuan.h> /* malloc hooks */
diff --git gnupg-2.2.8/sm/gpgsm.h.orig gnupg-2.2.8/sm/gpgsm.h
index 325948a..f55b55c 100644
--- gnupg-2.2.8/sm/gpgsm.h.orig
+++ gnupg-2.2.8/sm/gpgsm.h
@@ -51,7 +51,16 @@ struct keyserver_spec
 };
 
 
+#ifndef EXTERN_UNLESS_MAIN_MODULE
+#if !defined (INCLUDED_BY_MAIN_MODULE)
+#define EXTERN_UNLESS_MAIN_MODULE extern
+#else
+#define EXTERN_UNLESS_MAIN_MODULE
+#endif
+#endif
+
 /* A large struct named "opt" to keep global flags. */
+EXTERN_UNLESS_MAIN_MODULE
 struct
 {
   unsigned int debug; /* debug flags (DBG_foo_VALUE) */
diff --git gnupg-2.2.8/agent/agent.h.orig gnupg-2.2.8/agent/agent.h
index cf50d92..1ed912b 100644
--- gnupg-2.2.8/agent/agent.h.orig
+++ gnupg-2.2.8/agent/agent.h
@@ -54,7 +54,16 @@
 #define MAX_PASSPHRASE_LEN 255
 
 
-/* A large struct name "opt" to keep global flags */
+#ifndef EXTERN_UNLESS_MAIN_MODULE
+#if !defined (INCLUDED_BY_MAIN_MODULE)
+#define EXTERN_UNLESS_MAIN_MODULE extern
+#else
+#define EXTERN_UNLESS_MAIN_MODULE
+#endif
+#endif
+
+/* A large struct named "opt" to keep global flags. */
+EXTERN_UNLESS_MAIN_MODULE
 struct
 {
   unsigned int debug;  /* Debug flags (DBG_foo_VALUE) */
diff --git gnupg-2.2.8/agent/gpg-agent.c.orig gnupg-2.2.8/agent/gpg-agent.c
index 1fdc94d..4ee8d9e 100644
--- gnupg-2.2.8/agent/gpg-agent.c.orig
+++ gnupg-2.2.8/agent/gpg-agent.c
@@ -50,6 +50,7 @@
 #include <npth.h>
 
 #define GNUPG_COMMON_NEED_AFLOCAL
+#define INCLUDED_BY_MAIN_MODULE 1
 #include "agent.h"
 #include <assuan.h> /* Malloc hooks  and socket wrappers. */
 
diff --git gnupg-2.2.8/agent/protect-tool.c.orig gnupg-2.2.8/agent/protect-tool.c
index ec7b476..3bba5c6 100644
--- gnupg-2.2.8/agent/protect-tool.c.orig
+++ gnupg-2.2.8/agent/protect-tool.c
@@ -38,6 +38,7 @@
 #include <fcntl.h> /* for setmode() */
 #endif
 
+#define INCLUDED_BY_MAIN_MODULE 1
 #include "agent.h"
 #include "../common/i18n.h"
 #include "../common/get-passphrase.h"
diff --git gnupg-2.2.8/agent/preset-passphrase.c.orig gnupg-2.2.8/agent/preset-passphrase.c
index 7a9ea1b..3b40259 100644
--- gnupg-2.2.8/agent/preset-passphrase.c.orig
+++ gnupg-2.2.8/agent/preset-passphrase.c
@@ -44,6 +44,7 @@
 # include <windows.h>  /* To initialize the sockets.  fixme */
 #endif
 
+#define INCLUDED_BY_MAIN_MODULE 1
 #include "agent.h"
 #include "../common/simple-pwquery.h"
 #include "../common/i18n.h"
diff --git gnupg-2.2.8/agent/t-protect.c.orig gnupg-2.2.8/agent/t-protect.c
index d17c193..88b5525 100644
--- gnupg-2.2.8/agent/t-protect.c.orig
+++ gnupg-2.2.8/agent/t-protect.c
@@ -23,6 +23,7 @@
 #include <stdlib.h>
 #include <assert.h>
 
+#define INCLUDED_BY_MAIN_MODULE 1
 #include "agent.h"
 
 
diff --git gnupg-2.2.8/scd/scdaemon.h.orig gnupg-2.2.8/scd/scdaemon.h
index 4797f3d..f34d6c0 100644
--- gnupg-2.2.8/scd/scdaemon.h.orig
+++ gnupg-2.2.8/scd/scdaemon.h
@@ -43,7 +43,16 @@
 
 
 
-/* A large struct name "opt" to keep global flags. */
+#ifndef EXTERN_UNLESS_MAIN_MODULE
+#if !defined (INCLUDED_BY_MAIN_MODULE)
+#define EXTERN_UNLESS_MAIN_MODULE extern
+#else
+#define EXTERN_UNLESS_MAIN_MODULE
+#endif
+#endif
+
+/* A large struct named "opt" to keep global flags. */
+EXTERN_UNLESS_MAIN_MODULE
 struct
 {
   unsigned int debug; /* Debug flags (DBG_foo_VALUE). */
diff --git gnupg-2.2.8/scd/scdaemon.c.orig gnupg-2.2.8/scd/scdaemon.c
index 8f8a026..59a3dd9 100644
--- gnupg-2.2.8/scd/scdaemon.c.orig
+++ gnupg-2.2.8/scd/scdaemon.c
@@ -38,6 +38,7 @@
 #include <npth.h>
 
 #define GNUPG_COMMON_NEED_AFLOCAL
+#define INCLUDED_BY_MAIN_MODULE 1
 #include "scdaemon.h"
 #include <ksba.h>
 #include <gcrypt.h>
diff --git gnupg-2.2.8/dirmngr/dirmngr.h.orig gnupg-2.2.8/dirmngr/dirmngr.h
index 5189f93..e201734 100644
--- gnupg-2.2.8/dirmngr/dirmngr.h.orig
+++ gnupg-2.2.8/dirmngr/dirmngr.h
@@ -74,7 +74,16 @@ struct fingerprint_list_s
 };
 
 
-/* A large struct named "opt" to keep global flags.  */
+#ifndef EXTERN_UNLESS_MAIN_MODULE
+#if !defined (INCLUDED_BY_MAIN_MODULE)
+#define EXTERN_UNLESS_MAIN_MODULE extern
+#else
+#define EXTERN_UNLESS_MAIN_MODULE
+#endif
+#endif
+
+/* A large struct named "opt" to keep global flags. */
+EXTERN_UNLESS_MAIN_MODULE
 struct
 {
   unsigned int debug; /* debug flags (DBG_foo_VALUE) */
diff --git gnupg-2.2.8/dirmngr/dirmngr.c.orig gnupg-2.2.8/dirmngr/dirmngr.c
index 31f8e0f..70880dc 100644
--- gnupg-2.2.8/dirmngr/dirmngr.c.orig
+++ gnupg-2.2.8/dirmngr/dirmngr.c
@@ -56,6 +56,7 @@
 
 
 #define GNUPG_COMMON_NEED_AFLOCAL
+#define INCLUDED_BY_MAIN_MODULE 1
 #include "dirmngr.h"
 
 #include <assuan.h>
diff --git gnupg-2.2.8/tools/gpgtar.h.orig gnupg-2.2.8/tools/gpgtar.h
index 8cbe80b..4f67961 100644
--- gnupg-2.2.8/tools/gpgtar.h.orig
+++ gnupg-2.2.8/tools/gpgtar.h
@@ -23,7 +23,16 @@
 #include "../common/util.h"
 #include "../common/strlist.h"
 
-/* We keep all global options in the structure OPT.  */
+#ifndef EXTERN_UNLESS_MAIN_MODULE
+#if !defined (INCLUDED_BY_MAIN_MODULE)
+#define EXTERN_UNLESS_MAIN_MODULE extern
+#else
+#define EXTERN_UNLESS_MAIN_MODULE
+#endif
+#endif
+
+/* A large struct named "opt" to keep global flags. */
+EXTERN_UNLESS_MAIN_MODULE
 struct
 {
   int verbose;
diff --git gnupg-2.2.8/tools/gpgtar.c.orig gnupg-2.2.8/tools/gpgtar.c
index 2757ab0..a0ccc4a 100644
--- gnupg-2.2.8/tools/gpgtar.c.orig
+++ gnupg-2.2.8/tools/gpgtar.c
@@ -41,6 +41,7 @@
 #include "../common/init.h"
 #include "../common/strlist.h"
 
+#define INCLUDED_BY_MAIN_MODULE 1
 #include "gpgtar.h"
 
 
diff --git gnupg-2.2.8/tools/gpg-wks.h.orig gnupg-2.2.8/tools/gpg-wks.h
index 1b91b65..71be7ad 100644
--- gnupg-2.2.8/tools/gpg-wks.h.orig
+++ gnupg-2.2.8/tools/gpg-wks.h
@@ -29,7 +29,16 @@
 #define WKS_DRAFT_VERSION 3
 
 
-/* We keep all global options in the structure OPT.  */
+#ifndef EXTERN_UNLESS_MAIN_MODULE
+#if !defined (INCLUDED_BY_MAIN_MODULE)
+#define EXTERN_UNLESS_MAIN_MODULE extern
+#else
+#define EXTERN_UNLESS_MAIN_MODULE
+#endif
+#endif
+
+/* A large struct named "opt" to keep global flags. */
+EXTERN_UNLESS_MAIN_MODULE
 struct
 {
   int verbose;
diff --git gnupg-2.2.8/tools/gpg-wks-client.c.orig gnupg-2.2.8/tools/gpg-wks-client.c
index 73945ff..f48b116 100644
--- gnupg-2.2.8/tools/gpg-wks-client.c.orig
+++ gnupg-2.2.8/tools/gpg-wks-client.c
@@ -37,6 +37,7 @@
 #include "call-dirmngr.h"
 #include "mime-maker.h"
 #include "send-mail.h"
+#define INCLUDED_BY_MAIN_MODULE 1
 #include "gpg-wks.h"
 
 
diff --git gnupg-2.2.8/tools/gpgconf.h.orig gnupg-2.2.8/tools/gpgconf.h
index 8a061ef..ef72aa6 100644
--- gnupg-2.2.8/tools/gpgconf.h.orig
+++ gnupg-2.2.8/tools/gpgconf.h
@@ -22,7 +22,16 @@
 
 #include "../common/util.h"
 
-/* We keep all global options in the structure OPT.  */
+#ifndef EXTERN_UNLESS_MAIN_MODULE
+#if !defined (INCLUDED_BY_MAIN_MODULE)
+#define EXTERN_UNLESS_MAIN_MODULE extern
+#else
+#define EXTERN_UNLESS_MAIN_MODULE
+#endif
+#endif
+
+/* A large struct named "opt" to keep global flags. */
+EXTERN_UNLESS_MAIN_MODULE
 struct
 {
   int verbose;		/* Verbosity level.  */
diff --git gnupg-2.2.8/tools/gpgconf.c.orig gnupg-2.2.8/tools/gpgconf.c
index 59085d8..1f73e6f 100644
--- gnupg-2.2.8/tools/gpgconf.c.orig
+++ gnupg-2.2.8/tools/gpgconf.c
@@ -25,6 +25,7 @@
 #include <string.h>
 #include <unistd.h>
 
+#define INCLUDED_BY_MAIN_MODULE 1
 #include "gpgconf.h"
 #include "../common/i18n.h"
 #include "../common/sysutils.h"
