diff --git portage-2.3.1/bin/ebuild-helpers/keepdir portage-fixed/bin/ebuild-helpers/keepdir
index e0c852d..9779880 100755
--- portage-2.3.1/bin/ebuild-helpers/keepdir
+++ portage-fixed/bin/ebuild-helpers/keepdir
@@ -11,10 +11,5 @@ fi
 dodir "$@"
 ret=$?
 
-for x in "$@"; do
-	>> "${ED%/}/${x#/}/.keep_${CATEGORY}_${PN}-${SLOT%/*}" || \
-		{ echo "!!! ${0##*/}: cannot write .keep in ${ED%/}/${x#/}" 1>&2; ret=1; }
-done
-
 [[ ${ret} -ne 0 ]] && __helpers_die "${0##*/} failed"
 exit ${ret}
