--- portage-2.3.51/bin/phase-helpers.sh
+++ portage-2.3.51/bin/phase-helpers.sh
@@ -672,6 +672,23 @@ econf() {
 			fi
 		fi
 
+		# if the ebuild passes in --includedir, they're responsible for the conf_incdir fun.
+		local CONF_INCDIR INCDIR_VAR="INCDIR_${ABI}"
+		if [[ -n ${ABI} && -n ${!INCDIR_VAR} ]] ; then
+			CONF_INCDIR=${!INCDIR_VAR}
+		fi
+		if [[ -n ${CONF_INCDIR} ]] && ! __hasgq --includedir=\* "$@" ; then
+			export CONF_PREFIX=$(__hasg --exec-prefix=\* "$@")
+			[[ -z ${CONF_PREFIX} ]] && CONF_PREFIX=$(__hasg --prefix=\* "$@")
+			: ${CONF_PREFIX:=${EPREFIX}/usr}
+			CONF_PREFIX=${CONF_PREFIX#*=}
+			[[ ${CONF_PREFIX} != /* ]] && CONF_PREFIX="/${CONF_PREFIX}"
+			[[ ${CONF_INCDIR} != /* ]] && CONF_INCDIR="/${CONF_INCDIR}"
+			conf_args+=(
+				--includedir="$(__strip_duplicate_slashes "${CONF_PREFIX}${CONF_INCDIR}")"
+			)
+		fi
+
 		# if the profile defines a location to install libs to aside from default, pass it on.
 		# if the ebuild passes in --libdir, they're responsible for the conf_libdir fun.
 		local CONF_LIBDIR LIBDIR_VAR="LIBDIR_${ABI}"
@@ -977,6 +994,15 @@ best_version() {
 }
 
 if ___eapi_has_get_libdir; then
+	get_incdir() {
+		local incdir_var="INCDIR_${ABI}"
+		local incdir="include"
+
+		[[ -n ${ABI} && -n ${!incdir_var} ]] && incdir=${!incdir_var}
+
+		echo "${incdir}"
+	}
+
 	get_libdir() {
 		local libdir_var="LIBDIR_${ABI}"
 		local libdir="lib"
