--- portage-3.0.28/lib/_emerge/BinpkgExtractorAsync.py
+++ portage-3.0.28/lib/_emerge/BinpkgExtractorAsync.py
@@ -34,12 +34,7 @@ class BinpkgExtractorAsync(SpawnProcess):
             )
             output = process.communicate()[0]
             if b"--xattrs" in output:
-                tar_options = ["--xattrs", "--xattrs-include='*'"]
-                for x in portage.util.shlex_split(
-                    self.env.get("PORTAGE_XATTR_EXCLUDE", "")
-                ):
-                    tar_options.append(portage._shell_quote("--xattrs-exclude=%s" % x))
-                tar_options = " ".join(tar_options)
+                tar_options = "--xattrs"
 
         decomp = _compressors.get(compression_probe(self.pkg_path))
         if decomp is not None:
--- portage-3.0.28/lib/portage/dbapi/vartree.py
+++ portage-3.0.28/lib/portage/dbapi/vartree.py
@@ -1092,7 +1092,7 @@ class vardbapi(dbapi):
 
         opts, args = parser.parse_known_args(opts_list)
 
-        tar_cmd = ("tar", "-x", "--xattrs", "--xattrs-include=*", "-C", dest_dir)
+        tar_cmd = ("tar", "-xf", "-", "--xattrs", "-C", dest_dir)
         pr, pw = os.pipe()
         proc = await asyncio.create_subprocess_exec(*tar_cmd, stdin=pr)
         os.close(pr)
