--- portage-2.3.51/lib/portage/util/env_update.py
+++ portage-2.3.51/lib/portage/util/env_update.py
@@ -310,7 +310,7 @@ def _env_update(makelinks, target_root, prev_mtimes, contents, env,
 		settings["CHOST"] != settings["CBUILD"]:
 		ldconfig = find_binary("%s-ldconfig" % settings["CHOST"])
 	else:
-		ldconfig = os.path.join(eroot, "sbin", "ldconfig")
+		ldconfig = os.path.join(eroot, "usr", "bin", "ldconfig")
 
 	if ldconfig is None:
 		pass
