--- portage-3.0.20/bin/install-qa-check.d/90gcc-warnings
+++ portage-3.0.20/bin/install-qa-check.d/90gcc-warnings
@@ -27,8 +27,4 @@ gcc_warn_check() {
 			'warning: .*mathematical meaning*\[-Wparentheses\]'
 			# null argument where non-null required
 			'warning: .*\[-Wnonnull\]'
-			# array subscript is above/below/outside array bounds
-			'warning: .*\[-Warray-bounds\]'
-			# attempt to free a non-heap object
-			'warning: .*\[-Wfree-nonheap-object\]'
 			# those three do not have matching -W flags, it seems
@@ -43,16 +41,11 @@ gcc_warn_check() {
 			'warning: .*NULL.*\[-Wpointer-arith\]'
 			# pointer to a function used in arithmetic and likes
 			'warning: .*function.*\[-Wpointer-arith\]'
-			# the address of ... will never be NULL and likes
-			# (uses of function refs & string constants in conditionals)
-			'warning: .*\[-Waddress\]'
 			# outdated?
 			'warning: .*too few arguments for format'
 			# format ... expects a matching ... argument
 			# (iow, too few arguments for format in new wording :))
 			'warning: .*matching.*\[-Wformat=\]'
-			# function returns address of local variable
-			'warning: .*\[-Wreturn-local-addr\]'
 			# argument to sizeof ... is the same expression as the source
 			'warning: .*\[-Wsizeof-pointer-memaccess\]'
 			# iteration invokes undefined behavior
