diff --git efivar-30/src/makeguids.c.orig efivar-30/src/makeguids.c
index da2f5f7..452ed46 100644
--- efivar-30/src/makeguids.c.orig
+++ efivar-30/src/makeguids.c
@@ -152,6 +152,14 @@ main(int argc, char *argv[])
 	fprintf(symout, "#include <efivar.h>\n");
 	fprintf(symout, "#include <endian.h>\n");
 	fprintf(symout, """\n\
+#ifndef __bswap_constant_16\n\
+#define __bswap_constant_16(n) \\\n\
+  ((uint16_t)((((n) & 0xff00) >> 8) | (((n) & 0x00ff) << 8)))\n\
+#endif\n\
+#ifndef __bswap_constant_32\n\
+#define __bswap_constant_32(n) \\\n\
+  ((uint32_t)((((n) & 0xff000000) >> 24) | (((n) & 0x00ff0000) >>  8) | (((n) & 0x0000ff00) <<  8) | (((n) & 0x000000ff) << 24)))\n\
+#endif\n\
 #if BYTE_ORDER == BIG_ENDIAN\n\
 #define cpu_to_be32(n) (n)\n\
 #define cpu_to_be16(n) (n)\n\
