diff --git chromium-81.0.4044.113/chrome/updater/configurator.cc chromium-81.0.4044.113/chrome/updater/configurator.cc
index 00b74c2f9e..0f06989279 100644
--- chromium-81.0.4044.113/chrome/updater/configurator.cc
+++ chromium-81.0.4044.113/chrome/updater/configurator.cc
@@ -60,7 +60,7 @@ int Configurator::UpdateDelay() const {
 }
 
 std::vector<GURL> Configurator::UpdateUrl() const {
-  return std::vector<GURL>{GURL(kUpdaterJSONDefaultUrl)};
+  return std::vector<GURL>;
 }
 
 std::vector<GURL> Configurator::PingUrl() const {
