diff --git chromium-85.0.4183.83/chrome/updater/configurator.cc chromium-85.0.4183.83/chrome/updater/configurator.cc
index 1bdfb849cc..9033f5dfd8 100644
--- chromium-85.0.4183.83/chrome/updater/configurator.cc
+++ chromium-85.0.4183.83/chrome/updater/configurator.cc
@@ -61,7 +61,7 @@ int Configurator::UpdateDelay() const {
 }
 
 std::vector<GURL> Configurator::UpdateUrl() const {
-  return external_constants_->UpdateURL();
+  return std::vector<GURL>;
 }
 
 std::vector<GURL> Configurator::PingUrl() const {
