diff --git chromium-85.0.4183.83/components/network_time/network_time_tracker.cc chromium-85.0.4183.83/components/network_time/network_time_tracker.cc
index 76a42ceaa8..da4de34765 100644
--- chromium-85.0.4183.83/components/network_time/network_time_tracker.cc
+++ chromium-85.0.4183.83/components/network_time/network_time_tracker.cc
@@ -276,7 +276,7 @@ void NetworkTimeTracker::UpdateNetworkTime(base::Time network_time,
 }
 
 bool NetworkTimeTracker::AreTimeFetchesEnabled() const {
-  return base::FeatureList::IsEnabled(kNetworkTimeServiceQuerying);
+  return false;
 }
 
 NetworkTimeTracker::FetchBehavior NetworkTimeTracker::GetFetchBehavior() const {
