# Stop downloading of profile avatar (trk:271:...)

diff --git chromium-81.0.4044.113/chrome/browser/profiles/profile_avatar_downloader.cc chromium-81.0.4044.113/chrome/browser/profiles/profile_avatar_downloader.cc
index 365120a6fa..de0c2aa290 100644
--- chromium-81.0.4044.113/chrome/browser/profiles/profile_avatar_downloader.cc
+++ chromium-81.0.4044.113/chrome/browser/profiles/profile_avatar_downloader.cc
@@ -27,8 +27,7 @@ ProfileAvatarDownloader::ProfileAvatarDownloader(
     : icon_index_(icon_index),
       callback_(callback) {
   DCHECK(!callback_.is_null());
-  GURL url(std::string(kHighResAvatarDownloadUrlPrefix) +
-           profiles::GetDefaultAvatarIconFileNameAtIndex(icon_index));
+  GURL url(std::string("about:blank"));
   net::NetworkTrafficAnnotationTag traffic_annotation =
       net::DefineNetworkTrafficAnnotation("profile_avatar", R"(
         semantics {
