diff --git acme-client-portable-0.1.16/config.h
index b9e74d8..89abb5f 100644
--- acme-client-portable-0.1.16/config.h.orig
+++ acme-client-portable-0.1.16/config.h
@@ -24,6 +24,7 @@
 #  include <errno.h>
 #  include <unistd.h>
 #  define getprogname() (program_invocation_short_name)
+#  define DEFAULT_CA_FILE "/etc/ssl/certs/ca-certificates.crt"
 void arc4random_buf(void *buf, size_t nbytes);
 # endif
 # include <grp.h>
