diff --git libtirpc-1.0.1/tirpc/rpcsvc/nis.h.orig libtirpc-1.0.1/tirpc/rpcsvc/nis.h
--- libtirpc-1.0.1/tirpc/rpcsvc/nis.h.orig
+++ libtirpc-1.0.1/tirpc/rpcsvc/nis.h
@@ -36,7 +36,13 @@
 #include <rpc/rpc.h>
 #include <rpcsvc/nis_tags.h>
 
-__BEGIN_DECLS
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#ifndef __THROW
+#define __THROW
+#endif
 
 /*
  *	nis.h
@@ -611,6 +613,8 @@ extern  nis_error * nis_rmdir_3_svc (nis_name *, struct svc_req *) __THROW;
 extern  nis_error * nis_updkeys_3 (nis_name *, CLIENT *) __THROW;
 extern  nis_error * nis_updkeys_3_svc (nis_name *, struct svc_req *) __THROW;
 
-__END_DECLS
+#ifdef __cplusplus
+}
+#endif
 
 #endif /* ! _RPCSVC_NIS_H */
diff --git libtirpc-1.0.1/tirpc/rpcsvc/nislib.h.orig libtirpc-1.0.1/tirpc/rpcsvc/nislib.h
--- libtirpc-1.0.1/tirpc/rpcsvc/nislib.h.orig
+++ libtirpc-1.0.1/tirpc/rpcsvc/nislib.h
@@ -23,7 +23,13 @@
 
 #include <features.h>
 
-__BEGIN_DECLS
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#ifndef __THROW
+#define __THROW
+#endif
 
 typedef const char *const_nis_name;
 
@@ -283,6 +285,8 @@ extern nis_error __nisfind_server (const_nis_name, int, directory_obj **,
 
 #endif
 
-__END_DECLS
+#ifdef __cplusplus
+}
+#endif
 
 #endif	/* __RPCSVC_NISLIB_H__ */
