diff --git libtirpc-build/src/rpc_com.h.orig libtirpc-build/src/rpc_com.h
index 10bec79..41daad4 100644
--- libtirpc-build/src/rpc_com.h.orig
+++ libtirpc-build/src/rpc_com.h
@@ -61,8 +61,8 @@ void __xprt_unregister_unlocked(SVCXPRT *);
 void __xprt_set_raddr(SVCXPRT *, const struct sockaddr_storage *);
 
 
-SVCXPRT **__svc_xports;
-int __svc_maxrec;
+extern SVCXPRT **__svc_xports;
+extern int __svc_maxrec;
 
 #ifdef __cplusplus
 }
diff --git libtirpc-build/src/svc.c.orig libtirpc-build/src/svc.c
index b59467b..b5b0b42 100644
--- libtirpc-build/src/svc.c.orig
+++ libtirpc-build/src/svc.c
@@ -81,6 +81,9 @@ static void __xprt_do_unregister (SVCXPRT * xprt, bool_t dolock);
 
 /* ***************  SVCXPRT related stuff **************** */
 
+SVCXPRT **__svc_xports;
+int __svc_maxrec;
+
 /*
  * Activate a transport handle.
  */
