diff --git libtirpc-1.0.1/src/Makefile.am.orig libtirpc-1.0.1/src/Makefile.am
index e4ed8aa..fba2aa4 100644
--- libtirpc-1.0.1/src/Makefile.am.orig
+++ libtirpc-1.0.1/src/Makefile.am
@@ -24,7 +24,7 @@ libtirpc_la_SOURCES = auth_none.c auth_unix.c authunix_prot.c bindresvport.c cln
         rpcb_st_xdr.c svc.c svc_auth.c svc_dg.c svc_auth_unix.c svc_auth_none.c \
 	svc_auth_des.c \
         svc_generic.c svc_raw.c svc_run.c svc_simple.c svc_vc.c getpeereid.c \
-        auth_time.c auth_des.c authdes_prot.c debug.c
+        auth_time.c auth_des.c authdes_prot.c debug.c des_crypt.c des_impl.c
 
 ## XDR
 libtirpc_la_SOURCES += xdr.c xdr_rec.c xdr_array.c xdr_float.c xdr_mem.c xdr_reference.c xdr_stdio.c xdr_sizeof.c
