diff --git libtirpc-1.0.1/src/des_impl.c.orig libtirpc-1.0.1/src/des_impl.c
--- libtirpc-1.0.1/src/des_impl.c.orig
+++ libtirpc-1.0.1/src/des_impl.c
@@ -587,7 +587,7 @@ _des_crypt (char *buf, unsigned len, struct desparams *desp)
     }
   tin0 = tin1 = tout0 = tout1 = xor0 = xor1 = 0;
   tbuf[0] = tbuf[1] = 0;
-  __bzero (schedule, sizeof (schedule));
+  memset (schedule, 0, sizeof (schedule));
 
   return (1);
 }
diff --git libtirpc-1.0.1/tirpc/rpc/des.h.orig libtirpc-1.0.1/tirpc/rpc/des.h
--- libtirpc-1.0.1/tirpc/rpc/des.h.orig
+++ libtirpc-1.0.1/tirpc/rpc/des.h
@@ -46,14 +46,14 @@ enum desmode { CBC, ECB };
  * parameters to ioctl call
  */
 struct desparams {
-	u_char des_key[8];	/* key (with low bit parity) */
+	unsigned char des_key[8];	/* key (with low bit parity) */
 	enum desdir des_dir;	/* direction */
 	enum desmode des_mode;	/* mode */
-	u_char des_ivec[8];	/* input vector */
+	unsigned char des_ivec[8];	/* input vector */
 	unsigned des_len;	/* number of bytes to crypt */
 	union {
-		u_char UDES_data[DES_QUICKLEN];
-		u_char *UDES_buf;
+		unsigned char UDES_data[DES_QUICKLEN];
+		unsigned char *UDES_buf;
 	} UDES;
 #	define des_data UDES.UDES_data	/* direct data here if quick */
 #	define des_buf	UDES.UDES_buf	/* otherwise, pointer to data */
@@ -82,6 +82,6 @@ struct desparams {
 /*
  * Software DES.
  */
-extern int _des_crypt( char *, int, struct desparams * );
+extern int _des_crypt( char *, unsigned int, struct desparams * );
 
 #endif
