--- xorg-server-1.20.7/hw/xwayland/xwayland.c
+++ xorg-server-1.20.7/hw/xwayland/xwayland.c
@@ -40,7 +40,6 @@
 
 #ifdef XF86VIDMODE
 #include <X11/extensions/xf86vmproto.h>
-_X_EXPORT Bool noXFree86VidModeExtension;
+extern Bool noXFree86VidModeExtension;
 #endif
 
 void
--- xorg-server-1.20.7/test/misc.c
+++ xorg-server-1.20.7/test/misc.c
@@ -33,7 +33,7 @@
 
 #include "tests-common.h"
 
-ScreenInfo screenInfo;
+extern ScreenInfo screenInfo;
 
 static void
 dix_version_compare(void)
--- xorg-server-1.20.7/test/xi2/protocol-common.c
+++ xorg-server-1.20.7/test/xi2/protocol-common.c
@@ -44,6 +44,7 @@ WindowRec window;
 static ClientRec server_client;
 
 void *global_userdata;
+void (*reply_handler) (ClientPtr client, int len, char *data, void *userdata);
 
 int enable_GrabButton_wrap = 1;
 int enable_XISetEventMask_wrap = 1;
--- xorg-server-1.20.7/test/xi2/protocol-common.h
+++ xorg-server-1.20.7/test/xi2/protocol-common.h
@@ -99,7 +99,7 @@ extern void *global_userdata;
  * The reply handler called from WriteToClient. Set this handler if you need
  * to check the reply values.
  */
-void (*reply_handler) (ClientPtr client, int len, char *data, void *userdata);
+extern void (*reply_handler) (ClientPtr client, int len, char *data, void *userdata);
 
 /**
  * The default screen used for the windows. Initialized by init_simple().
