diff --git i3-4.13/i3bar/include/common.h i3-fixed/i3bar/include/common.h
index 0929e40..54c634c 100644
--- i3-4.13/i3bar/include/common.h
+++ i3-fixed/i3bar/include/common.h
@@ -17,7 +17,7 @@
 
 typedef struct rect_t rect;
 
-struct ev_loop *main_loop;
+extern struct ev_loop *main_loop;
 
 struct rect_t {
     int x;
diff --git i3-4.13/i3bar/include/configuration.h i3-fixed/i3bar/include/configuration.h
index c7c1f5e..914880a 100644
--- i3-4.13/i3bar/include/configuration.h
+++ i3-fixed/i3bar/include/configuration.h
@@ -62,7 +62,7 @@ typedef struct config_t {
            S_SHOW = 1 } hidden_state;
 } config_t;
 
-config_t config;
+extern config_t config;
 
 /**
  * Start parsing the received bar configuration JSON string
diff --git i3-4.13/i3bar/include/outputs.h i3-fixed/i3bar/include/outputs.h
index 3067581..e5b9205 100644
--- i3-4.13/i3bar/include/outputs.h
+++ i3-fixed/i3bar/include/outputs.h
@@ -19,7 +19,7 @@
 typedef struct i3_output i3_output;
 
 SLIST_HEAD(outputs_head, i3_output);
-struct outputs_head* outputs;
+extern struct outputs_head* outputs;
 
 /*
  * Start parsing the received JSON string
diff --git i3-4.13/i3bar/include/xcb.h i3-fixed/i3bar/include/xcb.h
index 7783e87..9907dfb 100644
--- i3-4.13/i3bar/include/xcb.h
+++ i3-fixed/i3bar/include/xcb.h
@@ -53,7 +53,7 @@ struct xcb_color_strings_t {
 typedef struct xcb_colors_t xcb_colors_t;
 
 /* Cached width of the custom separator if one was set */
-int separator_symbol_width;
+extern int separator_symbol_width;
 
 /*
  * Early initialization of the connection to X11: Everything which does not
diff --git i3-4.13/i3bar/src/config.c i3-fixed/i3bar/src/config.c
index d065ff4..5236063 100644
--- i3-4.13/i3bar/src/config.c
+++ i3-fixed/i3bar/src/config.c
@@ -19,6 +19,7 @@
 
 #include <X11/Xlib.h>
 
+config_t config;
 static char *cur_key;
 static bool parsing_bindings;
 static bool parsing_tray_outputs;
diff --git i3-4.13/i3bar/src/main.c i3-fixed/i3bar/src/main.c
index be684fc..33516f8 100644
--- i3-4.13/i3bar/src/main.c
+++ i3-fixed/i3bar/src/main.c
@@ -17,6 +17,8 @@
 #include <getopt.h>
 #include <glob.h>
 
+struct ev_loop *main_loop;
+
 /*
  * Having verboselog(), errorlog() and debuglog() is necessary when using libi3.
  *
diff --git i3-4.13/i3bar/src/outputs.c i3-fixed/i3bar/src/outputs.c
index 480c00b..a7687dd 100644
--- i3-4.13/i3bar/src/outputs.c
+++ i3-fixed/i3bar/src/outputs.c
@@ -17,6 +17,8 @@
 #include <yajl/yajl_parse.h>
 #include <yajl/yajl_version.h>
 
+struct outputs_head* outputs;
+
 /* A datatype to pass through the callbacks to save the state */
 struct outputs_json_params {
     struct outputs_head *outputs;
diff --git i3-4.13/i3bar/src/xcb.c i3-fixed/i3bar/src/xcb.c
index 26d83fc..2e4e5ee 100644
--- i3-4.13/i3bar/src/xcb.c
+++ i3-fixed/i3bar/src/xcb.c
@@ -77,6 +77,9 @@ xcb_colormap_t colormap;
 /* Overall height of the bar (based on font size) */
 int bar_height;
 
+/* Cached width of the custom separator if one was set */
+int separator_symbol_width;
+
 /* These are only relevant for XKB, which we only need for grabbing modifiers */
 int xkb_base;
 int mod_pressed = 0;
