diff --git i3-4.13/libi3/resolve_tilde.c i3-fixed/libi3/resolve_tilde.c
index 35770dc..3687c0b 100644
--- i3-4.13/libi3/resolve_tilde.c
+++ i3-fixed/libi3/resolve_tilde.c
@@ -19,27 +19,15 @@ void debuglog(char *fmt, ...) {
  *
  */
 char *resolve_tilde(const char *path) {
-    static glob_t globbuf;
-    char *head, *tail, *result;
-
-    tail = strchr(path, '/');
-    head = sstrndup(path, tail ? (size_t)(tail - path) : strlen(path));
-
-    int res = glob(head, GLOB_TILDE, NULL, &globbuf);
-    free(head);
-    /* no match, or many wildcard matches are bad */
-    if (res == GLOB_NOMATCH || globbuf.gl_pathc != 1)
-        result = sstrdup(path);
-    else if (res != 0) {
-        err(EXIT_FAILURE, "glob() failed");
-    } else {
-        head = globbuf.gl_pathv[0];
-        result = scalloc(strlen(head) + (tail ? strlen(tail) : 0) + 1, 1);
-        strncpy(result, head, strlen(head));
-        if (tail)
-            strncat(result, tail, strlen(tail));
-    }
-    globfree(&globbuf);
+    char *home, *result;
+
+    if (strncmp(path, "~/", 2))
+        return strdup(path);
+
+    if(!(home = getenv("HOME")))
+        err(1, "can't determine homedir");
+
+    sasprintf(&result, "%s%s", home, path+1);
 
     return result;
 }
diff --git i3-4.10.4/i3bar/src/main.c i3-fixed/i3bar/src/main.c
index a9a972d..43ae12a 100644
--- i3-4.10.4/i3bar/src/main.c
+++ i3-fixed/i3bar/src/main.c
@@ -45,13 +45,17 @@ void debuglog(char *fmt, ...) {
  *
  */
 char *expand_path(char *path) {
-    static glob_t globbuf;
-    if (glob(path, GLOB_NOCHECK | GLOB_TILDE, NULL, &globbuf) < 0) {
-        ELOG("glob() failed\n");
-        exit(EXIT_FAILURE);
-    }
-    char *result = sstrdup(globbuf.gl_pathc > 0 ? globbuf.gl_pathv[0] : path);
-    globfree(&globbuf);
+    char *home, *result;
+
+    if (strncmp(path, "~/", 2))
+        return strdup(path);
+
+    if(!(home = getenv("HOME")))
+        ELOG("can't determine homedir\n");
+        exit(EXIT_FAILURE);
+
+    sasprintf(&result, "%s/%s", getenv("HOME"), path+2);
+
     return result;
 }
 
