diff -ur pulseaudio-6.0/configure.ac pulseaudio-fixed/configure.ac
--- pulseaudio-6.0/configure.ac	2015-02-12 14:10:35.000000000 +0000
+++ pulseaudio-fixed/configure.ac	2015-10-04 15:06:02.443497958 +0000
@@ -748,7 +748,7 @@
     AS_HELP_STRING([--disable-oss-wrapper],[Disable optional OSS wrapper support]))
 
 AS_IF([test "x$enable_oss_output" != "xno" -o "x$enable_oss_wrapper" != "xno"],
-    [AC_CHECK_HEADERS([sys/soundcard.h], HAVE_OSS=1, HAVE_OSS=0)],
+    [AC_CHECK_HEADERS([sys/soundcard.h], HAVE_OSS=0, HAVE_OSS=0)],
     HAVE_OSS=0)
 
 AS_IF([test "x$enable_oss_output" = "xyes" -o "x$enable_oss_wrapper" = "xyes" && test "x$HAVE_OSS" = "x0"],
