diff --git pulseaudio-9.0/configure.ac pulseaudio-fixed/configure.ac
index 4edc8e0..ac4586f 100644
--- pulseaudio-9.0/configure.ac
+++ pulseaudio-fixed/configure.ac
@@ -778,7 +778,7 @@ AC_ARG_ENABLE([oss-wrapper],
     AS_HELP_STRING([--disable-oss-wrapper],[Disable optional OSS wrapper support]))
 
 AS_IF([test "x$enable_oss_output" != "xno" -o "x$enable_oss_wrapper" != "xno"],
-    [AC_CHECK_HEADERS([sys/soundcard.h], HAVE_OSS=1, HAVE_OSS=0)],
+    [AC_CHECK_HEADERS([sys/soundcard.h], HAVE_OSS=0, HAVE_OSS=0)],
     HAVE_OSS=0)
 
 AS_IF([test "x$enable_oss_output" = "xyes" -o "x$enable_oss_wrapper" = "xyes" && test "x$HAVE_OSS" = "x0"],
