--- wget-1.19.5/tests/unit-tests.c
+++ wget-1.19.5/tests/unit-tests.c
@@ -36,8 +36,6 @@ as that of the covered work.  */
 
 #include "unit-tests.h"
 
-const char *program_argstring = "TEST";
-
 static int tests_run;
 
 static const char *
@@ -68,8 +66,6 @@ all_tests(void)
   return NULL;
 }
 
-const char *program_name; /* Needed by lib/error.c. */
-
 int
 main (int argc _GL_UNUSED, const char *argv[])
 {
@@ -85,6 +81,7 @@ main (int argc _GL_UNUSED, const char *argv[])
 #endif /* ENABLE_NLS */
 
   program_name = argv[0];
+  program_argstring = "TEST";
 
   result = all_tests();
 
