diff --git WireGuard-0.0.20170421/src/data.c
index 63cd60e..bc5d1f9 100644
--- WireGuard-0.0.20170421/src/data.c.orig
+++ WireGuard-0.0.20170421/src/data.c
@@ -235,7 +235,6 @@ static void begin_parallel_encryption(struct padata_priv *padata)
 {
 	struct encryption_ctx *ctx = container_of(padata, struct encryption_ctx, padata);
 	queue_encrypt_reset(&ctx->queue, ctx->keypair);
-	padata_do_serial(padata);
 }
 
 static void finish_parallel_encryption(struct padata_priv *padata)
@@ -363,7 +362,6 @@ static void begin_parallel_decryption(struct padata_priv *padata)
 {
 	struct decryption_ctx *ctx = container_of(padata, struct decryption_ctx, padata);
 	begin_decrypt_packet(ctx);
-	padata_do_serial(padata);
 }
 
 static void finish_parallel_decryption(struct padata_priv *padata)
