diff --git valgrind-3.14.0/VEX/priv/guest_s390_defs.h
index 3bfecbe..3aafdf6 100644
--- valgrind-3.14.0/VEX/priv/guest_s390_defs.h.orig
+++ valgrind-3.14.0/VEX/priv/guest_s390_defs.h
@@ -264,7 +264,7 @@ extern ULong last_execute_target;
 /* Vector operatons passed to s390x_dirtyhelper_vec_op(...) helper.
    Please don't change ordering of elements and append new items
    before  S390_VEC_OP_LAST. */
-enum {
+typedef enum {
    S390_VEC_OP_INVALID = 0,
    S390_VEC_OP_VPKS = 1,
    S390_VEC_OP_VPKLS = 2,
