--- valgrind-3.13.0/memcheck/tests/str_tester.c
+++ valgrind-3.13.0/memcheck/tests/str_tester.c
@@ -504,7 +504,7 @@ test_strchrnul (void)
 #endif
 
 // DDD: better done by testing for the function.
-#if !defined(__APPLE__) && !defined(__sun)
+#if defined(__GLIBC__)
 static void
 test_rawmemchr (void)
 {
@@ -1442,7 +1442,7 @@ main (void)
   test_strchrnul ();
 # endif
 
-# if !defined(__APPLE__) && !defined(__sun)
+# if defined(__GLIBC__)
   /* rawmemchr.  */
   test_rawmemchr ();
 # endif
--- valgrind-3.13.0/memcheck/tests/linux/Makefile.am
+++ valgrind-3.13.0/memcheck/tests/linux/Makefile.am
@@ -31,7 +31,6 @@ check_PROGRAMS = \
 	lsframe2 \
 	rfcomm \
 	sigqueue \
-	stack_changes \
 	stack_switch \
 	syscalls-2007 \
 	syslog-syscall \
--- valgrind-3.13.0/memcheck/tests/linux/sigqueue.c
+++ valgrind-3.13.0/memcheck/tests/linux/sigqueue.c
@@ -22,7 +22,7 @@ int main(int argc, char **argv)
     fprintf(stdout, "offsetof(siginfo_t, si_code)   = %zd\n",
             offsetof(siginfo_t, si_code));
     fprintf(stdout, "offsetof(siginfo_t, _sifields) = %zd\n",
-            offsetof(siginfo_t, _sifields));
+            offsetof(siginfo_t, __si_fields));
   }
   si = calloc(1, sz);
   si->si_signo = SIGWINCH;
--- valgrind-3.13.0/memcheck/tests/amd64-linux/int3-amd64.c
+++ valgrind-3.13.0/memcheck/tests/amd64-linux/int3-amd64.c
@@ -32,7 +32,7 @@ int main(void) {
    puts("main");
    register_handler(SIGTRAP, int_handler);
    asm volatile(
-      "movabsq $zz_int, %%rdx\n"
+      "leaq zz_int(%%rip), %%rdx\n"
       "mov %%rdx, %0\n"
       "zz_int:\n"
       "int $3\n"
--- valgrind-3.13.0/none/tests/pth_atfork1.c
+++ valgrind-3.13.0/none/tests/pth_atfork1.c
@@ -18,7 +18,7 @@
    Boston, MA 02111-1307, USA.  */
 
 #include <errno.h>
-#if !defined(__APPLE__) && !defined(__sun)
+#if defined(__GLIBC__)
 # include <error.h>
 #endif
 #include <stdlib.h>
@@ -27,7 +27,7 @@
 #include <sys/wait.h>
 #include <stdio.h>
 
-#if defined(__APPLE__) || defined(__sun)
+#if !defined(__GLIBC__)
 #include <string.h>  /* strerror */
 static void error (int status, int errnum, char* msg)
 {
--- valgrind-3.13.0/helgrind/tests/tc20_verifywrap.c
+++ valgrind-3.13.0/helgrind/tests/tc20_verifywrap.c
@@ -20,7 +20,7 @@
 
 #if !defined(__APPLE__)
 
-#if defined(__sun__)
+#if !defined(__GLIBC_PREREQ)
 /* Fake __GLIBC_PREREQ on Solaris. Pretend glibc >= 2.4. */
 # define __GLIBC_PREREQ
 #else
--- valgrind-3.13.0/drd/tests/pth_detached3.c
+++ valgrind-3.13.0/drd/tests/pth_detached3.c
@@ -21,7 +21,7 @@ int main(int argc, char** argv)
   pthread_detach(thread);
 
   /* Invoke pthread_detach() with an invalid thread ID. */
-  pthread_detach(thread + 8);
+  pthread_detach((pthread_t)((char *)thread + 8));
 
   fprintf(stderr, "Finished.\n");
 
diff --git valgrind-3.14.0/memcheck/tests/linux/sys-statx.c
index fe9f9ba..d175b19 100644
--- valgrind-3.14.0/memcheck/tests/linux/sys-statx.c.orig
+++ valgrind-3.14.0/memcheck/tests/linux/sys-statx.c
@@ -7,6 +7,9 @@
 #include <assert.h>
 #include <string.h>
 #include <sys/syscall.h>
+#ifndef __GLIBC_PREREQ
+#define __GLIBC_PREREQ(...) 0
+#endif
 #if __GLIBC_PREREQ(2,28)
 /* struct statx provided in sys/stat.h */
 #else
