From bc15ad5784a0e653cdc07ec9c1dbd2ea2e45f12e Mon Sep 17 00:00:00 2001
From: Michael Drake <michael.drake@codethink.co.uk>
Date: Mon, 15 Feb 2021 14:24:15 +0000
Subject: tests: Update corestring allocation count.

---
 test/corestrings.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/test/corestrings.c b/test/corestrings.c
index aaa23e4d0..43eb5130e 100644
--- a/test/corestrings.c
+++ b/test/corestrings.c
@@ -40,7 +40,7 @@
  *
  * This is used to test all the out of memory paths in initialisation.
  */
-#define CORESTRING_TEST_COUNT 483
+#define CORESTRING_TEST_COUNT 484
 
 START_TEST(corestrings_test)
 {
-- 
cgit v1.2.1

