--- Python-3.10.3/Modules/posixmodule.c
+++ Python-3.10.3/Modules/posixmodule.c
@@ -126,10 +126,11 @@ corresponding Unix manual entries for more information on calls.");
 #undef HAVE_SCHED_SETAFFINITY
 #endif
 
-#if defined(HAVE_SYS_XATTR_H) && defined(__GLIBC__) && !defined(__FreeBSD_kernel__) && !defined(__GNU__)
+#if defined(HAVE_SYS_XATTR_H) && defined(__linux__)
 #  define USE_XATTRS
 #endif
 
 #ifdef USE_XATTRS
+#  include <linux/limits.h>
 #  include <sys/xattr.h>
 #endif
 
