diff --git a/libcxx-3.8.1.src/include/__mutex_base b/libcxx-fixed/include/__mutex_base
index b019b47..0dbe62b 100644
--- a/libcxx-3.8.1.src/include/__mutex_base
+++ b/libcxx-fixed/include/__mutex_base
@@ -32,11 +32,7 @@ class _LIBCPP_TYPE_VIS mutex
 
 public:
     _LIBCPP_INLINE_VISIBILITY
-#ifndef _LIBCPP_HAS_NO_CONSTEXPR
-     constexpr mutex() _NOEXCEPT : __m_(PTHREAD_MUTEX_INITIALIZER) {}
-#else
      mutex() _NOEXCEPT {__m_ = (pthread_mutex_t)PTHREAD_MUTEX_INITIALIZER;}
-#endif
      ~mutex();
 
 private:
@@ -271,11 +267,7 @@ class _LIBCPP_TYPE_VIS condition_variable
     pthread_cond_t __cv_;
 public:
     _LIBCPP_INLINE_VISIBILITY
-#ifndef _LIBCPP_HAS_NO_CONSTEXPR
-    constexpr condition_variable() : __cv_(PTHREAD_COND_INITIALIZER) {}
-#else
     condition_variable() {__cv_ = (pthread_cond_t)PTHREAD_COND_INITIALIZER;}
-#endif
     ~condition_variable();
 
 private:
