diff --git a/src/os.cpp b/src/os.cpp
index 60c66908..8b919aeb 100644
--- a/src/os.cpp
+++ b/src/os.cpp
@@ -2096,6 +2096,10 @@ const char *possible_ld_names[] = {
 #if defined(ZIG_ARCH_X86_64)
     "ld-linux-x86-64.so.2",
     "ld-musl-x86_64.so.1",
+#elif defined(ZIG_ARCH_POWERPC64)
+    "ld-linux-powerpc64.so.2",
+    "ld-musl-powerpc64.so.1",
+    "ld-musl-powerpc64-ld128.so.1",
 #elif defined(ZIG_ARCH_ARM64)
     "ld-linux-aarch64.so.1",
     "ld-musl-aarch64.so.1",
diff --git a/src/os.hpp b/src/os.hpp
index b301937e..375bf4e6 100644
--- a/src/os.hpp
+++ b/src/os.hpp
@@ -31,7 +31,9 @@
 #define ZIG_OS_UNKNOWN
 #endif
 
-#if defined(__x86_64__)
+#if defined(__powerpc64__)
+#define ZIG_ARCH_POWERPC64
+#elif defined(__x86_64__)
 #define ZIG_ARCH_X86_64
 #elif defined(__aarch64__)
 #define ZIG_ARCH_ARM64
