diff --git pinentry-0.9.5/pinentry/Makefile.am pinentry-fixed/pinentry/Makefile.am
index be99822..a7e67f5 100644
--- pinentry-0.9.5/pinentry/Makefile.am
+++ pinentry-fixed/pinentry/Makefile.am
@@ -36,7 +36,7 @@ endif
 noinst_LIBRARIES = libpinentry.a $(pinentry_curses)
 
 LDADD = $(COMMON_LIBS)
-AM_CPPFLAGS = $(COMMON_CFLAGS) -I$(top_srcdir)/secmem
+AM_CPPFLAGS = $(COMMON_CFLAGS) $(NCURSES_INCLUDE) -I$(top_srcdir)/secmem
 
 libpinentry_a_SOURCES = pinentry.h pinentry.c argparse.c argparse.h \
 	password-cache.h password-cache.c $(pinentry_emacs_sources)
