diff --git uniconvertor-2.0_pre379/src/uc2/formats/sk1/sk1objs/curvelow.c uniconvertor-fixed/src/uc2/formats/sk1/sk1objs/curvelow.c
index 5e5e2e7..0f26220 100644
--- uniconvertor-2.0_pre379/src/uc2/formats/sk1/sk1objs/curvelow.c
+++ uniconvertor-fixed/src/uc2/formats/sk1/sk1objs/curvelow.c
@@ -23,7 +23,8 @@
  */
 
 
-#include "math.h"
+#include <math.h>
+#include <stdlib.h>
 // #include "X11/Xlib.h"
 #include "curvelow.h"
 
