diff --git dhcpcd-6.11.3/common.c dhcpcd-6.11.3/common.c
index 8c6060c..9e32a0b 100644
--- dhcpcd-6.11.3/common.c
+++ dhcpcd-6.11.3/common.c
@@ -178,6 +178,7 @@ logger(struct dhcpcd_ctx *ctx, int pri, const char *fmt, ...)
 #endif
 		vfprintf(pri <= LOG_ERR ? stderr : stdout, fmt, vac);
 		fputc('\n', pri <= LOG_ERR ? stderr : stdout);
+		fflush(pri <= LOG_ERR ? stderr : stdout);
 		va_end(vac);
 	}
 
