--- compiler-rt-7.0.1.src/lib/builtins/CMakeLists.txt
+++ compiler-rt-7.0.1.src/lib/builtins/CMakeLists.txt
@@ -514,6 +514,9 @@ set(powerpc64_SOURCES
   ppc/gcc_qsub.c
   ppc/multc3.c
   ${GENERIC_SOURCES})
+list(FILTER powerpc64_SOURCES EXCLUDE REGEX ".*tf.*")
+list(FILTER powerpc64_SOURCES EXCLUDE REGEX ".*tc.*")
+list(FILTER powerpc64_SOURCES EXCLUDE REGEX ".*gcc_q.*")
 set(powerpc64le_SOURCES ${powerpc64_SOURCES})
 
 set(riscv_SOURCES ${GENERIC_SOURCES} ${GENERIC_TF_SOURCES})
