diff --git hdf5-1.8.18/src/H5Dio.c
index 58031e8..fbf8034 100644
--- hdf5-1.8.18/src/H5Dio.c.orig
+++ hdf5-1.8.18/src/H5Dio.c
@@ -77,6 +77,7 @@ static herr_t H5D__ioinfo_term(H5D_io_info_t *io_info);
 #endif /* H5_HAVE_PARALLEL */
 static herr_t H5D__typeinfo_term(const H5D_type_info_t *type_info);
 
+herr_t H5T_patch_vlen_file(H5T_t *dt, H5F_t *f);
 
 /*********************/
 /* Package Variables */
