diff --git cmake-3.14.3/Modules/FindPkgConfig.cmake.orig cmake-3.14.3/Modules/FindPkgConfig.cmake
index e192426..10c5cb8 100644
--- cmake-3.14.3/Modules/FindPkgConfig.cmake.orig
+++ cmake-3.14.3/Modules/FindPkgConfig.cmake
@@ -343,7 +343,7 @@ macro(_pkg_check_modules_internal _is_required _is_silent _no_cmake_path _no_cma
       if(NOT DEFINED CMAKE_SYSTEM_NAME
           OR (CMAKE_SYSTEM_NAME MATCHES "^(Linux|kFreeBSD|GNU)$"
               AND NOT CMAKE_CROSSCOMPILING))
-        if(EXISTS "/etc/debian_version") # is this a debian system ?
+        if(CMAKE_LIBRARY_ARCHITECTURE)
           if(CMAKE_LIBRARY_ARCHITECTURE)
             list(APPEND _lib_dirs "lib/${CMAKE_LIBRARY_ARCHITECTURE}/pkgconfig")
           endif()
