diff --git lttng-ust-2.7.1/include/lttng/ust-dlfcn.h
index c92f1bb..a6f6824 100644
--- lttng-ust-2.7.1/include/lttng/ust-dlfcn.h.orig
+++ lttng-ust-2.7.1/include/lttng/ust-dlfcn.h
@@ -52,6 +52,10 @@
 #undef dlclose
 #undef dlopen
 
+#ifndef __nonnull
+#define __nonnull(x) __attribute__((__nonnull__(x)))
+#endif
+
 extern void *dlopen(__const char *__file, int __mode);
 extern int dlclose(void *__handle) __nonnull ((1));
 extern void *dlsym(void *__restrict __handle,
