diff --git lttng-ust-2.7.1/include/usterr-signal-safe.h.orig lttng-ust-2.7.1/include/usterr-signal-safe.h
index bc68fbf4b2..940bd18d36 100644
--- lttng-ust-2.7.1/include/usterr-signal-safe.h.orig
+++ lttng-ust-2.7.1/include/usterr-signal-safe.h
@@ -106,7 +106,7 @@ do {									\
 #define ERR(fmt, args...)	ERRMSG("Error: " fmt, ## args)
 #define BUG(fmt, args...)	ERRMSG("BUG: " fmt, ## args)
 
-#if !defined(__linux__) || ((_POSIX_C_SOURCE >= 200112L || _XOPEN_SOURCE >= 600) && !defined(_GNU_SOURCE))
+#if !defined(__linux__) || ((_POSIX_C_SOURCE >= 200112L || _XOPEN_SOURCE >= 600) && !defined(_GNU_SOURCE)) || !defined(__GLIBC__)
 /*
  * Version using XSI strerror_r.
  */
