--- llpp-33/wsi/x11/wsi.ml
+++ llpp-33/wsi/x11/wsi.ml
@@ -1146,9 +1146,7 @@ let init t w h platform =
         let addr =
           match platform with
           | Utils.Pmacos -> Unix.ADDR_UNIX d
-          | Utils.Plinux ->
-             Unix.ADDR_UNIX ("\000/tmp/.X11-unix/X" ^ string_of_int dispnum)
-          | Utils.Punknown | Utils.Pbsd ->
+          | Utils.Plinux | Utils.Punknown | Utils.Pbsd ->
              Unix.ADDR_UNIX ("/tmp/.X11-unix/X" ^ string_of_int dispnum)
         in
         Unix.socket Unix.PF_UNIX Unix.SOCK_STREAM 0, addr
