diff --git boinc-client_release-7.8-7.8.1/lib/crypt.h
index 63c577a..64a9b35 100644
--- boinc-client_release-7.8-7.8.1/lib/crypt.h.orig
+++ boinc-client_release-7.8-7.8.1/lib/crypt.h
@@ -27,9 +27,11 @@
 #include <openssl/rsa.h>
 
 #if (OPENSSL_VERSION_NUMBER >= 0x10100000L) /* OpenSSL 1.1.0+ */
+#ifndef LIBRESSL_VERSION_NUMBER
 #define HAVE_OPAQUE_EVP_PKEY 1 /* since 1.1.0 -pre3 */
 #define HAVE_OPAQUE_RSA_DSA_DH 1 /* since 1.1.0 -pre5 */
 #endif
+#endif
 
 #define MAX_RSA_MODULUS_BITS 1024
 #define MAX_RSA_MODULUS_LEN ((MAX_RSA_MODULUS_BITS + 7) / 8)
