--- i3-4.13/i3-nagbar/i3-nagbar.h
+++ i3-fixed/i3-nagbar/i3-nagbar.h
@@ -13,7 +13,7 @@
         }                      \
     } while (0)
 
-#define xmacro(atom) xcb_atom_t A_##atom;
+#define xmacro(atom) extern xcb_atom_t A_##atom;
 #include "atoms.xmacro"
 #undef xmacro
 
--- i3-4.13/i3-nagbar/main.c
+++ i3-fixed/i3-nagbar/main.c
@@ -34,6 +34,10 @@
 
 #include "i3-nagbar.h"
 
+#define xmacro(atom) xcb_atom_t A_##atom;
+#include "atoms.xmacro"
+#undef xmacro
+
 /** This is the equivalent of XC_left_ptr. I’m not sure why xcb doesn’t have a
  * constant for that. */
 #define XCB_CURSOR_LEFT_PTR 68
