diff --git ddcutil-0.7.1/src/util/edid.h
index 633cef2..ee753c1 100644
--- ddcutil-0.7.1/src/util/edid.h.orig
+++ ddcutil-0.7.1/src/util/edid.h
@@ -76,7 +76,7 @@ struct {
    char         marker[4];          // always "EDID"
    Byte         bytes[128];
    char         mfg_id[EDID_MFG_ID_FIELD_SIZE];
-   ushort       model_hex;
+   unsigned short model_hex;
    char         model_name[EDID_MODEL_NAME_FIELD_SIZE];                   // tag 0xfc
    uint32_t     serial_binary;
    char         serial_ascii[EDID_SERIAL_ASCII_FIELD_SIZE];               // tag 0xff
@@ -85,11 +85,11 @@ struct {
    bool         is_model_year;   // if true, year is model year, if false, is manufacture year
    Byte         edid_version_major;
    Byte         edid_version_minor;
-   ushort       wx;
-   ushort       wy;
-   ushort       rx, ry;
-   ushort       gx, gy;
-   ushort       bx, by;
+   unsigned short wx;
+   unsigned short wy;
+   unsigned short rx, ry;
+   unsigned short gx, gy;
+   unsigned short bx, by;
    char *       edid_source;
 } Parsed_Edid;
 
