--- cfe-7.0.1.src/lib/Driver/ToolChains/Clang.cpp
+++ cfe-7.0.1.src/lib/Driver/ToolChains/Clang.cpp
@@ -1603,7 +1603,9 @@ void Clang::AddPPCTargetArgs(const ArgList &Args,
                              ArgStringList &CmdArgs) const {
   // Select the ABI to use.
   const char *ABIName = nullptr;
-  if (getToolChain().getTriple().isOSLinux())
+  if (getToolChain().getTriple().isMusl())
+    ABIName = "elfv2";
+  else if (getToolChain().getTriple().isOSLinux())
     switch (getToolChain().getArch()) {
     case llvm::Triple::ppc64: {
       // When targeting a processor that supports QPX, or if QPX is
