diff --git libcxx-3.9.1.src/include/cstddef.orig libcxx-3.9.1.src/include/cstddef
--- libcxx-3.9.1.src/include/cstddef.orig
+++ libcxx-3.9.1.src/include/cstddef
@@ -48,7 +48,7 @@ _LIBCPP_BEGIN_NAMESPACE_STD
 using ::ptrdiff_t;
 using ::size_t;
 
-#if defined(__CLANG_MAX_ALIGN_T_DEFINED) || defined(_GCC_MAX_ALIGN_T)
+#if defined(__CLANG_MAX_ALIGN_T_DEFINED) || defined(_GCC_MAX_ALIGN_T) || defined(__DEFINED_max_align_t)
 // Re-use the compiler's <stddef.h> max_align_t where possible.
 using ::max_align_t;
 #else
diff --git libcxx-3.9.1.src/include/stddef.h.orig libcxx-3.9.1.src/include/stddef.h
--- libcxx-3.9.1.src/include/stddef.h.orig
+++ libcxx-3.9.1.src/include/stddef.h
@@ -53,7 +53,7 @@ using std::nullptr_t;
 }
 
 // Re-use the compiler's <stddef.h> max_align_t where possible.
-#if !defined(__CLANG_MAX_ALIGN_T_DEFINED) && !defined(_GCC_MAX_ALIGN_T)
+#if !defined(__CLANG_MAX_ALIGN_T_DEFINED) && !defined(_GCC_MAX_ALIGN_T) && !defined(__DEFINED_max_align_t)
 typedef long double max_align_t;
 #endif
 
