--- sudo-1.8.28p1/lib/util/mksigname.h
+++ sudo-1.8.28p1/lib/util/mksigname.h
@@ -41,10 +41,6 @@ sudo_sys_signame[0] = "Signal 0";
     if (sudo_sys_signame[SIGKILL] == NULL)
 	sudo_sys_signame[SIGKILL] = "KILL";
 #endif
-#ifdef SIGUNUSED
-    if (sudo_sys_signame[SIGUNUSED] == NULL)
-	sudo_sys_signame[SIGUNUSED] = "UNUSED";
-#endif
 #ifdef SIGBUS
     if (sudo_sys_signame[SIGBUS] == NULL)
 	sudo_sys_signame[SIGBUS] = "BUS";
