diff --git man-db-2.7.5/src/man.c man-db-fixed/src/man.c
index bea684f..df5f5cb 100644
--- man-db-2.7.5/src/man.c
+++ man-db-fixed/src/man.c
@@ -1503,11 +1503,9 @@ static void add_output_iconv (pipeline *p,
 {
 	debug ("add_output_iconv: source %s, target %s\n", source, target);
 	if (source && target && !STREQ (source, target)) {
-		char *target_translit = xasprintf ("%s//TRANSLIT", target);
 		pipeline_command_args (p, "iconv", "-c",
-				       "-f", source, "-t", target_translit,
+				       "-f", source, "-t", target,
 				       NULL);
-		free (target_translit);
 	}
 }
 
diff --git man-db-2.7.5/src/manconv.c man-db-fixed/src/manconv.c
index cae9b67..48b7a81 100644
--- man-db-2.7.5/src/manconv.c
+++ man-db-fixed/src/manconv.c
@@ -106,10 +106,9 @@ static int try_iconv (pipeline *p, const char *try_from_code, const char *to,
 	static char *utf8 = NULL, *output = NULL;
 	size_t utf8left = 0;
 	iconv_t cd_utf8, cd = NULL;
-	int to_utf8 = STREQ (try_to_code, "UTF-8") ||
-		      STRNEQ (try_to_code, "UTF-8//", 7);
-	const char *utf8_target = last ? "UTF-8//IGNORE" : "UTF-8";
-	int ignore_errors = (strstr (try_to_code, "//IGNORE") != NULL);;
+	int to_utf8 = STREQ (try_to_code, "UTF-8");
+	const char *utf8_target = "UTF-8";
+	int ignore_errors = 0;
 	int ret = 0;
 
 	debug ("trying encoding %s -> %s\n", try_from_code, try_to_code);
diff --git man-db-2.7.5/src/manconv_client.c man-db-fixed/src/manconv_client.c
index 2949c75..7805f90 100644
--- man-db-2.7.5/src/manconv_client.c
+++ man-db-fixed/src/manconv_client.c
@@ -131,7 +131,7 @@ void add_manconv (pipeline *p, const char *source, const char *target)
 		codes->from[2] = NULL;
 		name = appendstr (name, "UTF-8:", source, NULL);
 	}
-	codes->to = xasprintf ("%s//IGNORE", target);
+	codes->to = xstrdup (target);
 	/* informational only; no shell quoting concerns */
 	name = appendstr (name, " -t ", codes->to, NULL);
 	if (quiet >= 2)
diff --git man-db-2.7.5/src/manconv_main.c man-db-fixed/src/manconv_main.c
index 166a2b8..4e700e9 100644
--- man-db-2.7.5/src/manconv_main.c
+++ man-db-fixed/src/manconv_main.c
@@ -105,9 +105,6 @@ static error_t parse_opt (int key, char *arg, struct argp_state *state)
 			return 0;
 		case 't':
 			to_code = xstrdup (arg);
-			if (!strstr (to_code, "//"))
-				to_code = appendstr (to_code, "//TRANSLIT",
-						     NULL);
 			return 0;
 		case 'd':
 			debug_level = 1;
diff --git man-db-2.7.5/src/whatis.c man-db-fixed/src/whatis.c
index 39aa093..0a25c92 100644
--- man-db-2.7.5/src/whatis.c
+++ man-db-fixed/src/whatis.c
@@ -948,7 +948,7 @@ int main (int argc, char *argv[])
 	display_seen = hashtable_create (&null_hashtable_free);
 
 #ifdef HAVE_ICONV
-	locale_charset = xasprintf ("%s//IGNORE", get_locale_charset ());
+	locale_charset = xstrdup (get_locale_charset ());
 	conv_to_locale = iconv_open (locale_charset, "UTF-8");
 	free (locale_charset);
 #endif /* HAVE_ICONV */
