diff --git pulseaudio-11.1/src/modules/raop/raop-crypto.c
index 0dd6463..8023c38 100644
--- pulseaudio-11.1/src/modules/raop/raop-crypto.c.orig
+++ pulseaudio-11.1/src/modules/raop/raop-crypto.c
@@ -48,7 +48,7 @@
  * be able to upgrade to the new openssl version. To insulate ourselves from
  * such transition problems, let's implement RSA_set0_key() ourselves if it's
  * not available. */
-#if OPENSSL_VERSION_NUMBER < 0x10100000L
+#if OPENSSL_VERSION_NUMBER < 0x10100000L || defined(LIBRESSL_VERSION_NUMBER)
 static int RSA_set0_key(RSA *r, BIGNUM *n, BIGNUM *e, BIGNUM *d) {
     r->n = n;
     r->e = e;
