- localename #ifdef soup doesn't handle musl
--- m4-1.4.17/tests/Makefile.in
+++ m4-1.4.17/tests/Makefile.in
@@ -133,7 +133,7 @@ TESTS = test-alloca-opt$(EXEEXT) test-array_list$(EXEEXT) \
 	$(am__EXEEXT_3) test-langinfo$(EXEEXT) test-link$(EXEEXT) \
 	test-linkedhash_list$(EXEEXT) $(am__EXEEXT_4) \
 	test-locale$(EXEEXT) test-localeconv$(EXEEXT) \
-	test-localename$(EXEEXT) test-lseek.sh test-lstat$(EXEEXT) \
+	test-lseek.sh test-lstat$(EXEEXT) \
 	test-malloc-gnu$(EXEEXT) test-malloca$(EXEEXT) $(am__EXEEXT_5) \
 	test-math$(EXEEXT) test-mbrtowc1.sh test-mbrtowc2.sh \
 	test-mbrtowc3.sh test-mbrtowc4.sh test-mbrtowc-w32-1.sh \
@@ -209,7 +209,7 @@ check_PROGRAMS = test-alloca-opt$(EXEEXT) test-array_list$(EXEEXT) \
 	test-langinfo$(EXEEXT) test-link$(EXEEXT) \
 	test-linkedhash_list$(EXEEXT) $(am__EXEEXT_4) \
 	test-locale$(EXEEXT) test-localeconv$(EXEEXT) \
-	test-localename$(EXEEXT) test-lseek$(EXEEXT) \
+	test-lseek$(EXEEXT) \
 	test-lstat$(EXEEXT) test-malloc-gnu$(EXEEXT) \
 	test-malloca$(EXEEXT) $(am__EXEEXT_5) test-math$(EXEEXT) \
 	test-mbrtowc$(EXEEXT) test-mbrtowc-w32$(EXEEXT) \
