diff --git i3-4.13/src/x.c i3-fixed/src/x.c
index 8d7d3dd..4982e0e 100644
--- i3-4.13/src/x.c
+++ i3-fixed/src/x.c
@@ -556,9 +556,9 @@ void x_draw_decoration(Con *con) {
 
         draw_util_text(title, &(parent->frame_buffer),
                        p->color->text, p->color->background,
-                       con->deco_rect.x + logical_px(2),
+                       con->deco_rect.x + logical_px(8),
                        con->deco_rect.y + text_offset_y,
-                       con->deco_rect.width - 2 * logical_px(2));
+                       con->deco_rect.width - logical_px(8));
         I3STRING_FREE(title);
 
         goto after_title;
@@ -602,9 +602,9 @@ void x_draw_decoration(Con *con) {
     i3String *title = con->title_format == NULL ? win->name : con_parse_title_format(con);
     draw_util_text(title, &(parent->frame_buffer),
                    p->color->text, p->color->background,
-                   con->deco_rect.x + logical_px(2),
+                   con->deco_rect.x + logical_px(8),
                    con->deco_rect.y + text_offset_y,
-                   con->deco_rect.width - mark_width - 2 * logical_px(2));
+                   con->deco_rect.width - mark_width - logical_px(8));
     if (con->title_format != NULL)
         I3STRING_FREE(title);
 
