diff --git i3-4.13/i3bar/src/xcb.c i3-fixed/i3bar/src/xcb.c
index 2715e44..26d83fc 100644
--- i3-4.13/i3bar/src/xcb.c
+++ i3-fixed/i3bar/src/xcb.c
@@ -465,7 +465,7 @@ void handle_button(xcb_button_press_event_t *event) {
 
     DLOG("Got button %d\n", event->detail);
 
-    int workspace_width = 0;
+    int workspace_width = logical_px(2);
     i3_ws *cur_ws = NULL, *clicked_ws = NULL, *ws_walk;
 
     TAILQ_FOREACH(ws_walk, walk->workspaces, tailq) {
