--- patchelf-0.10/src/patchelf.cc
+++ patchelf-0.10/src/patchelf.cc
@@ -745,8 +745,8 @@ void ElfFile<ElfFileParamNames>::rewriteSectionsLibrary()
     if (isExecutable) {
         if (startOffset >= startPage) {
             debug("shifting new PT_LOAD segment by %d bytes to work around a Linux kernel bug\n", startOffset - startPage);
+            startPage = startOffset;
         }
-        startPage = startOffset;
     }
 
     /* Add a segment that maps the replaced sections into memory. */
