diff --git gst-plugins-good-1.14.4/tests/check/elements/rtpsession.c.orig gst-plugins-good-1.14.4/tests/check/elements/rtpsession.c
index 8c19ce2..31b34e1 100644
--- gst-plugins-good-1.14.4/tests/check/elements/rtpsession.c.orig
+++ gst-plugins-good-1.14.4/tests/check/elements/rtpsession.c
@@ -436,7 +436,7 @@ GST_START_TEST (test_internal_sources_timeout)
   /* verify BYE and RR */
   j = 0;
   seen_bye = FALSE;
-  while (!seen_bye) {
+  for (i = 0; i < 5; i++) {
     session_harness_produce_rtcp (h, 1);
     buf = session_harness_pull_rtcp (h);
     fail_unless (buf != NULL);
@@ -468,6 +468,7 @@ GST_START_TEST (test_internal_sources_timeout)
     gst_rtcp_buffer_unmap (&rtcp);
     gst_buffer_unref (buf);
   }
+  fail_unless_equals_int (TRUE, seen_bye);
   fail_unless_equals_int (0x3, j); /* verify we got both BYE and RR */
 
   session_harness_free (h);
