--- tevent-0.9.35/wscript
+++ tevent-0.9.35/wscript
@@ -115,7 +115,7 @@ def build(bld):
         for env in bld.gen_python_environments(['PKGCONFIGDIR']):
             bld.SAMBA_PYTHON('_tevent',
                             'pytevent.c',
-                            deps='tevent',
+                            deps='replace tevent',
                             realname='_tevent.so',
                             cflags='-DPACKAGE_VERSION=\"%s\"' % VERSION)
 
