diff --git perf/bench/futex-hash.c
index 58ae6ed..1c63a71 100644
--- perf/bench/futex-hash.c.orig
+++ perf/bench/futex-hash.c
@@ -35,7 +35,7 @@ static unsigned int nfutexes = 1024;
 static bool fshared = false, done = false, silent = false;
 static int futex_flag = 0;
 
-struct timeval start, end, runtime;
+static struct timeval start, end, runtime;
 static pthread_mutex_t thread_lock;
 static unsigned int threads_starting;
 static struct stats throughput_stats;
diff --git perf/bench/futex-lock-pi.c
index 08653ae..10d908c 100644
--- perf/bench/futex-lock-pi.c.orig
+++ perf/bench/futex-lock-pi.c
@@ -34,7 +34,7 @@ static bool silent = false, multi = false;
 static bool done = false, fshared = false;
 static unsigned int ncpus, nthreads = 0;
 static int futex_flag = 0;
-struct timeval start, end, runtime;
+static struct timeval start, end, runtime;
 static pthread_mutex_t thread_lock;
 static unsigned int threads_starting;
 static struct stats throughput_stats;
