diff -ur gperf-3.0.4/src/output.cc gperf-fixed/src/output.cc
--- gperf-3.0.4/src/output.cc	2009-01-19 03:50:07.000000000 -0600
+++ gperf-fixed/src/output.cc	2016-01-30 12:16:44.404265088 -0600
@@ -1299,7 +1299,7 @@
 
           if (option[DEBUG])
             fprintf (stderr,
-                     "dup_ptr[%d]: hash_value = %d, index = %d, count = %d\n",
+                     "dup_ptr[%ld]: hash_value = %d, index = %d, count = %d\n",
                      dup_ptr - duplicates,
                      dup_ptr->hash_value, dup_ptr->index, dup_ptr->count);
 
