diff --git xbmc-16.1-Jarvis/xbmc/cores/VideoRenderers/RenderCapture.cpp.orig xbmc-16.1-Jarvis/xbmc/cores/VideoRenderers/RenderCapture.cpp
index fcd742b8d4..7bde00e3ee 100644
--- xbmc-16.1-Jarvis/xbmc/cores/VideoRenderers/RenderCapture.cpp.orig
+++ xbmc-16.1-Jarvis/xbmc/cores/VideoRenderers/RenderCapture.cpp
@@ -19,6 +19,7 @@
  */
 
 #include "RenderCapture.h"
+#include "rendering/RenderSystem.h"
 #include "utils/log.h"
 #include "windowing/WindowingFactory.h"
 #include "settings/AdvancedSettings.h"
