diff --git gegl-0.2.0/operations/external/ff-load.c.orig gegl-0.2.0/operations/external/ff-load.c
index 75d26e9..e78cefd 100644
--- gegl-0.2.0/operations/external/ff-load.c.orig
+++ gegl-0.2.0/operations/external/ff-load.c
@@ -321,7 +321,7 @@ prepare (GeglOperation *operation)
       p->width = p->enc->width;
       p->height = p->enc->height;
       p->frames = 10000000;
-      p->lavc_frame = avcodec_alloc_frame ();
+      p->lavc_frame = av_frame_alloc ();
 
       if (p->fourcc)
         g_free (p->fourcc);
