--- a/av1/encoder/rdopt.c
+++ b/av1/encoder/rdopt.c
@@ -8253,7 +8253,7 @@ static INLINE void get_this_mv(int_mv *this_mv, int this_mode, int ref_idx,
     this_mv->as_int = INVALID_MV;
   } else if (single_mode == GLOBALMV) {
     *this_mv = mbmi_ext->global_mvs[ref_frame[ref_idx]];
-  } else {
+  } else if (single_mode == NEARMV || single_mode == NEARESTMV) {
     assert(single_mode == NEARMV || single_mode == NEARESTMV);
     const int ref_mv_offset = get_ref_mv_offset(single_mode, ref_mv_idx);
     if (ref_mv_offset < mbmi_ext->ref_mv_count[ref_frame_type]) {
