diff --git llvm-3.9.0.src/tools/lldb/source/Host/common/FileSpec.cpp llvm-3.9.0-fixed/tools/lldb/source/Host/common/FileSpec.cpp
index 53c0ab4..e99121a 100644
--- llvm-3.9.0.src/tools/lldb/source/Host/common/FileSpec.cpp
+++ llvm-3.9.0-fixed/tools/lldb/source/Host/common/FileSpec.cpp
@@ -17,6 +17,7 @@
 #ifndef _MSC_VER
 #include <libgen.h>
 #endif
+#include <limits.h>
 #include <set>
 #include <string.h>
 #include <fstream>
diff --git llvm-3.9.0.src/tools/lldb/tools/lldb-mi/CMakeLists.txt llvm-3.9.0-fixed/tools/lldb/tools/lldb-mi/CMakeLists.txt
index 79f657a..2a80dec 100644
--- llvm-3.9.0.src/tools/lldb/tools/lldb-mi/CMakeLists.txt
+++ llvm-3.9.0-fixed/tools/lldb/tools/lldb-mi/CMakeLists.txt
@@ -84,7 +84,7 @@ endif ()
 
 add_lldb_executable(lldb-mi ${LLDB_MI_SOURCES})
 
-target_link_libraries(lldb-mi liblldb)
+target_link_libraries(lldb-mi liblldb LLVMSupport)
 if ( NOT CMAKE_SYSTEM_NAME MATCHES "Windows" )
   target_link_libraries(lldb-mi pthread)
 endif ()
