--- socat-1.7.3.1/xio-ip.c
+++ socat-fixed/xio-ip.c
@@ -384,7 +384,9 @@ int xiogetaddrinfo(const char *node, const char *service,
       /*!!! try gethostbyname2 for IP6 */
       if ((host = Gethostbyname(node)) == NULL) {
 	 Error2("gethostbyname(\"%s\"): %s", node,
+#ifdef NETDB_INTERNAL
 		h_errno == NETDB_INTERNAL ? strerror(errno) :
+#endif
 		hstrerror(h_errno));
 #if HAVE_RESOLV_H
 	 if (res_opts0 | res_opts1) {
--- socat-1.7.3.1/xio-ip4.c
+++ socat-fixed/xio-ip4.c
@@ -46,7 +46,9 @@ int xioparsenetwork_ip4(const char *rangename, struct xiorange *range) {
       if ((maskaddr = Gethostbyname(delimpos+1)) == NULL) {
 	 /* note: cast is req on AIX: */
 	 Error2("gethostbyname(\"%s\"): %s", delimpos+1,
+#ifdef NETDB_INTERNAL
 		h_errno == NETDB_INTERNAL ? strerror(errno) :
+#endif
 		(char *)hstrerror(h_errno));
 	 return STAT_NORETRY;
       }
@@ -62,7 +64,9 @@ int xioparsenetwork_ip4(const char *rangename, struct xiorange *range) {
       if ((nameaddr = Gethostbyname(rangename1)) == NULL) {
 	 /* note: cast is req on AIX: */
 	 Error2("gethostbyname(\"%s\"): %s", rangename1,
+#ifdef NETDB_INTERNAL
 		h_errno == NETDB_INTERNAL ? strerror(errno) :
+#endif
 		(char *)hstrerror(h_errno));
 	    free(rangename1);
 	 return STAT_NORETRY;
--- socat-1.7.3.1/xio-proxy.c
+++ socat-fixed/xio-proxy.c
@@ -246,7 +246,9 @@ int _xioopen_proxy_prepare(struct proxyvars *proxyvars, struct opt *opts,
 	 int level = E_WARN;
 	 /* note: cast is req on AIX: */
 	 Msg2(level, "gethostbyname(\"%s\"): %s", targetname,
+#ifdef NETDB_INTERNAL
 	      h_errno == NETDB_INTERNAL ? strerror(errno) :
+#endif
 	      (char *)hstrerror(h_errno)/*0 h_messages[h_errno-1]*/);
 
 	 proxyvars->targetaddr = strdup(targetname);
