diff --git ddcutil-0.8.4/src/util/edid.h
index 90f67f9..bebfe5e 100644
--- ddcutil-0.8.4/src/util/edid.h.orig
+++ ddcutil-0.8.4/src/util/edid.h
@@ -68,7 +68,7 @@ struct {
    char         marker[4];                                    ///< always "EDID"
    Byte         bytes[128];                                  ///< raw bytes of EDID
    char         mfg_id[EDID_MFG_ID_FIELD_SIZE];              ///< 3 character mfg id, null terminated
-   ushort       product_code;                                ///< product code number
+   unsigned short product_code;                              ///< product code number
    char         model_name[EDID_MODEL_NAME_FIELD_SIZE];      ///< model name (tag 0xfc)
    uint32_t     serial_binary;                               ///< binary serial number
    char         serial_ascii[EDID_SERIAL_ASCII_FIELD_SIZE];  ///< serial number string (tag 0xff)
@@ -77,14 +77,14 @@ struct {
    bool         is_model_year;           ///< if true, year is model year, if false, is manufacture year
    Byte         edid_version_major;      ///< EDID major version number
    Byte         edid_version_minor;      ///< EDID minor version number
-   ushort       wx;                      ///< whitepoint x coordinate
-   ushort       wy;                      ///< whitepoint y coordinate
-   ushort       rx;                      ///< red   x coordinate
-   ushort       ry;                      ///< red   y coordinate
-   ushort       gx;                      ///< green x coordinate
-   ushort       gy;                      ///< green y coordinate
-   ushort       bx;                      ///< blue  x coordinate
-   ushort       by;                      ///< blue  y coordinate
+   unsigned short wx;                    ///< whitepoint x coordinate
+   unsigned short wy;                    ///< whitepoint y coordinate
+   unsigned short rx;                    ///< red   x coordinate
+   unsigned short ry;                    ///< red   y coordinate
+   unsigned short gx;                    ///< green x coordinate
+   unsigned short gy;                    ///< green y coordinate
+   unsigned short bx;                    ///< blue  x coordinate
+   unsigned short by;                    ///< blue  y coordinate
    Byte         video_input_definition;  /// EDID byte 20 (x14)
    // bool         is_digital_input;      // from byte 20 (x14), but 7
    uint8_t      extension_flag;          ///< number of optional extension blocks
