diff --git ddcutil-0.8.6/src/base/core.h
index 1e96afc..38292ba 100644
--- ddcutil-0.8.6/src/base/core.h.orig
+++ ddcutil-0.8.6/src/base/core.h
@@ -327,8 +327,8 @@ void program_logic_error(
  *  Reports an error in program logic.
  * @ingroup abnormal_termination
  */
-#define PROGRAM_LOGIC_ERROR(format, ...) \
-   program_logic_error(__func__, __LINE__, __FILE__, format, ##__VA_ARGS__)
+#define PROGRAM_LOGIC_ERROR(...) \
+   program_logic_error(__func__, __LINE__, __FILE__, __VA_ARGS__)
 
 
 #ifdef OLD
