--- netsurf-3.8/frontends/gtk/download.h.orig
+++ netsurf-3.8/frontends/gtk/download.h
@@ -24,7 +24,7 @@
 /**
  * download operation table for gtk
  */
-struct gui_download_table *nsgtk_download_table;
+extern struct gui_download_table *nsgtk_download_table;
 
 /**
  * Initialise download window ready for use.
--- netsurf-3.8/frontends/gtk/fetch.h.orig
+++ netsurf-3.8/frontends/gtk/fetch.h
@@ -19,7 +19,7 @@
 #ifndef NETSURF_GTK_FETCH_H
 #define NETSURF_GTK_FETCH_H
 
-struct gui_fetch_table *nsgtk_fetch_table;
+extern struct gui_fetch_table *nsgtk_fetch_table;
 
 void gtk_fetch_filetype_init(const char *mimefile);
 void gtk_fetch_filetype_fin(void);
--- netsurf-3.8/frontends/gtk/search.h.orig
+++ netsurf-3.8/frontends/gtk/search.h
@@ -19,7 +19,7 @@
 #ifndef _NETSURF_GTK_SEARCH_H_
 #define _NETSURF_GTK_SEARCH_H_
 
-struct gui_search_table *nsgtk_search_table;
+extern struct gui_search_table *nsgtk_search_table;
 
 struct nsgtk_scaffolding;
 
--- netsurf-3.8/frontends/gtk/selection.h.orig
+++ netsurf-3.8/frontends/gtk/selection.h
@@ -19,6 +19,6 @@
 #ifndef GTK_SELECTION_H
 #define GTK_SELECTION_H
 
-struct gui_clipboard_table *nsgtk_clipboard_table;
+extern struct gui_clipboard_table *nsgtk_clipboard_table;
 
 #endif
diff --git netsurf-3.8/utils/file.h.orig netsurf-3.8/utils/file.h
index 5ee13b5..0282c35 100644
--- netsurf-3.8/utils/file.h.orig
+++ netsurf-3.8/utils/file.h
@@ -117,7 +117,7 @@ struct gui_file_table {
 };
 
 /** Default (posix) file operation table. */
-struct gui_file_table *default_file_table;
+extern struct gui_file_table *default_file_table;
 
 /**
  * Generate a path from one or more component elemnts.
