diff -ur numactl-2.0.11-orig/syscall.c numactl-2.0.11/syscall.c
--- numactl-2.0.11-orig/syscall.c	2015-12-10 15:24:17.000000000 -0600
+++ numactl-2.0.11/syscall.c	2016-02-29 20:49:05.130694048 -0600
@@ -115,7 +115,8 @@
 
 #endif
 
-#if defined(__GLIBC__) && __GLIBC_PREREQ(2, 11)
+#if defined(__GLIBC__) 
+#if __GLIBC_PREREQ(2, 11)
 
 /* glibc 2.11 seems to have working 6 argument sycall. Use the
    glibc supplied syscall in this case.
@@ -180,6 +181,9 @@
 #else
 #define syscall6 syscall
 #endif
+#else
+#define syscall6 syscall
+#endif
 
 long WEAK get_mempolicy(int *policy, const unsigned long *nmask,
 				unsigned long maxnode, void *addr, int flags)
