diff --git a/binutils/readelf.c b/binutils/readelf.c
index 166dd23..4b6f9fc 100644
--- a/binutils/readelf.c
+++ b/binutils/readelf.c
@@ -9639,7 +9639,7 @@ process_dynamic_section (Filedata * filedata)
 	  if (archive_file_offset != 0)
 	    str_tab_len = archive_file_size - offset;
 	  else
-	    str_tab_len = filedata->file_size;
+	    str_tab_len = filedata->file_size - offset;
 
 	  if (str_tab_len < 1)
 	    {
-- 
2.9.3

