diff --git iputils-67e7d0daf1f231cc708217e6aec2f8d5ce7aeacf/ping6_common.c
index 934742b..375206d 100644
--- iputils-67e7d0daf1f231cc708217e6aec2f8d5ce7aeacf/ping6_common.c.orig
+++ iputils-67e7d0daf1f231cc708217e6aec2f8d5ce7aeacf/ping6_common.c
@@ -96,7 +96,7 @@ static int pr_icmph(__u8 type, __u8 code, __u32 info);
 void ping6_usage(unsigned) __attribute((noreturn));
 
 struct sockaddr_in6 source6 = { .sin6_family = AF_INET6 };
-char *device;
+extern char *device;
 
 #if defined(USE_GCRYPT) || defined(USE_OPENSSL) || defined(USE_NETTLE)
 #include "iputils_md5dig.h"
