/bin might be a symlink
--- pytest-shutil-1.2.11/tests/integration/test_cmdline_integration.py
+++ pytest-shutil-1.2.11/tests/integration/test_cmdline_integration.py
@@ -5,6 +5,6 @@ from pytest_shutil import cmdline
 
 def test_chdir():
     here = os.getcwd()
-    with cmdline.chdir('/bin'):
-        assert os.getcwd() == '/bin'
+    with cmdline.chdir('/etc'):
+        assert os.getcwd() == '/etc'
     assert os.getcwd() == here
\ No newline at end of file
