--- kbd-2.0.4/src/version.h
+++ kbd-2.0.4/src/version.h
@@ -5,7 +5,7 @@
 #include <string.h>
 #include <stdlib.h>
 
-char *progname;
+extern char *progname;
 
 static inline void
 set_progname(char *name)
--- kbd-2.0.4/src/kbd_error.c
+++ kbd-2.0.4/src/kbd_error.c
@@ -9,7 +9,7 @@
 #include "kbd.h"
 #include "kbd_error.h"
 
-extern const char *progname;
+const char *progname;
 
 void
     __attribute__((format(printf, 2, 3)))
