diff -ur perl-5.20.2/pp_sys.c perl-5.20.2-fixed/pp_sys.c
--- perl-5.20.2/pp_sys.c	2015-01-11 08:03:10.000000000 -0600
+++ perl-5.20.2-fixed/pp_sys.c	2016-01-30 13:54:49.366062616 -0600
@@ -26,6 +26,9 @@
  * By 'system', we mean ops which interact with the OS, such as pp_open().
  */
 
+#define _GNU_SOURCE
+#include <unistd.h>
+
 #include "EXTERN.h"
 #define PERL_IN_PP_SYS_C
 #include "perl.h"
diff -ur perl-5.20.2/sv.c perl-5.20.2-fixed/sv.c
--- perl-5.20.2/sv.c	2015-01-15 16:12:06.000000000 -0600
+++ perl-5.20.2-fixed/sv.c	2016-01-30 13:55:14.353411799 -0600
@@ -27,6 +27,10 @@
  * in the pp*.c files.
  */
 
+
+#define _GNU_SOURCE
+#include <stdlib.h>
+
 #include "EXTERN.h"
 #define PERL_IN_SV_C
 #include "perl.h"
