--- cups-filters-1.20.4/filter/foomatic-rip/foomaticrip.c
+++ cups-filters-1.20.4/filter/foomatic-rip/foomaticrip.c
@@ -105,7 +105,7 @@ jobparams_t * get_current_job()
 }
 
 
-dstr_t *postpipe;  /* command into which the output of this filter should be piped */
+extern dstr_t *postpipe;  /* command into which the output of this filter should be piped */
 FILE *postpipe_fh = NULL;
 
 FILE * open_postpipe()
@@ -131,7 +131,7 @@ FILE * open_postpipe()
 }
 
 
-char printer_model[256] = "";
+extern char printer_model[256];
 char attrpath[256] = "";
 
 
@@ -148,7 +148,7 @@ int cm_disabled = 0;
 
 /* These variables were in 'dat' before */
 char colorprofile [128];
-char cupsfilter[256];
+extern char cupsfilter[256];
 char **jclprepend = NULL;
 dstr_t *jclappend;
 
--- cups-filters-1.20.4/filter/foomatic-rip/options.c
+++ cups-filters-1.20.4/filter/foomatic-rip/options.c
@@ -38,7 +38,7 @@ typedef struct icc_mapping_entry_s {
 } icc_mapping_entry_t;
 
 /* Values from foomatic keywords in the ppd file */
-char printer_model [256];
+char printer_model [256] = "";
 char printer_id [256];
 char driver [128];
 char cmd [4096];
